---
title: 'Extract, Analyze and Visualize Mutational Signatures with Sigminer'
author: 
  - Shixiang Wang, PhD (Sun Yat-sen University Cancer Center)
  - Xue-Song Liu, PhD (ShanghaiTech University)
date: "`r Sys.Date()`"
knit: "bookdown::render_book"
site: bookdown::bookdown_site
output:
  bookdown::bs4_book:
  bookdown::pdf_book:
    dev: "cairo_pdf"
    latex_engine: xelatex
bibliography: book.bib
biblio-style: apalike
always_allow_html: yes
toc_appendix: yes
fontsize: "12pt"
mainfont: Times New Roman
sansfont: Arial
monofontoptions: "Scale=0.7"
linestretch: 1.5
toc-depth: 2
link-citations: yes
github-repo: shixiangwang/sigminer-book
twiter-handle: WangShxiang
description: "Extract, analyze and visualize mutational signatures with sigminer."
documentclass: book
papersize: A4
classoption: twoside
highlight_bw: yes
geometry: "left=35mm,right=35mm,top=25mm,bottom=25mm"
---


```{r include=FALSE}
CRANpkg <- function (pkg) {
    cran <- "https://CRAN.R-project.org/package"
    fmt <- "[%s](%s=%s)"
    sprintf(fmt, pkg, cran, pkg)
}
Biocpkg <- function (pkg) {
    sprintf("[%s](http://bioconductor.org/packages/%s)", pkg, pkg)
}
Githubpkg <- function (user, pkg) {
    gh <- "https://github.com"
    fmt <- "[%s](%s/%s/%s)"
    sprintf(fmt, pkg, gh, user, pkg)
}
badge_version <- function(pkg, color="green", bioc = FALSE, user = NULL) {
    v <- packageVersion(pkg)
    url <- if (is.null(user)) {
      if (bioc) Biocpkg(pkg) else CRANpkg(pkg)
    } else {
      Githubpkg(user, pkg)
    }
    
    badger::badge_custom(pkg, v, color, url)
}


library(emojifont)
library(knitr)
opts_chunk$set(message=FALSE, warning=FALSE, eval=TRUE, 
    echo=TRUE, cache=TRUE, dev="CairoPNG", out.width="95%")
```


# `r emoji("book")` Introduction {-}

`r badge_version("UCSCXenaTools")`
`r badge_version("sigminer")`
`r badge_version("maftools", bioc = TRUE)`
[![Hits](https://hits.seeyoufarm.com/api/count/incr/badge.svg?url=https%3A%2F%2Fgithub.com%2FShixiangWang%2Fsigminer-book&count_bg=%2379C83D&title_bg=%23555555&icon=americanairlines.svg&icon_color=%23E7E7E7&title=hits&edge_flat=false)](https://hits.seeyoufarm.com)


## `r emoji("dart")` Motivation {-}

The book is written as a guide for extracting, analyzing and visualizing mutational signatures with R/CRAN package **sigminer**. The [*README*](https://github.com/ShixiangWang/sigminer/blob/master/README.md) and [*Reference list*](https://shixiangwang.github.io/sigminer/reference/index.html) of **sigminer** have given users overview and the very details of specific points (e.g., functions) in **sigminer**. This book will help users focus on quickly getting the mutational signature analysis done to make life easy. 

In this book, we assume you have already known how to operate [R](http://cran.r-project.org/).

## `r emoji("memo")` Citation {-}

If you use **sigminer** or its pipeline version **sigflow** in published research, please cite the most appropriate paper(s):

1. **Wang, S.**, Li, H., Song, M., Tao, Z., Wu, T., He, Z., ... & Liu, X. S. (2021). Copy number signature analysis tool and its application in prostate cancer reveals distinct mutational processes and clinical outcomes. ___PLoS genetics___, 17(5), e1009557. <https://doi.org/10.1371/journal.pgen.1009557>
2. **Wang, S.**, Tao, Z., Wu, T., & Liu, X. S. (2021). Sigflow: an automated and comprehensive pipeline for cancer genome mutational signature analysis. ___Bioinformatics___, 37(11), 1590-1592. <https://doi.org/10.1093/bioinformatics/btaa895>

Please properly cite the following references when you are using any
corresponding features. The references are also listed in the function
documentation. Very thanks to the works, **sigminer** cannot be created
without the giants.

1.  Mayakonda, Anand, et al. “Maftools: efficient and comprehensive
    analysis of somatic variants in cancer.” ___Genome research___ 28.11
    (2018): 1747-1756.
2.  Gaujoux, Renaud, and Cathal Seoighe. “A Flexible R Package for
    Nonnegative Matrix Factorization.”" ___BMC Bioinformatics___ 11, no. 1
    (December 2010).
3.  Wickham, Hadley. "ggplot2." ___Wiley Interdisciplinary Reviews: Computational Statistics___ 3.2 (2011): 180-185.
4.  Kim, Jaegil, et al. “Somatic ERCC2 mutations are associated with a
    distinct genomic signature in urothelial tumors.” ___Nature Genetics___
    48.6 (2016): 600.
5.  Alexandrov, Ludmil B., et al. “Deciphering signatures of mutational
    processes operative in human cancer.” ___Cell Reports___ 3.1 (2013):
    246-259.
6.  Degasperi, Andrea, et al. “A practical framework and online tool for
    mutational signature analyses show intertissue variation and driver
    dependencies.” ___Nature Cancer___ 1.2 (2020): 249-263.
7.  Alexandrov, Ludmil B., et al. “The repertoire of mutational
    signatures in human cancer.” ___Nature___ 578.7793 (2020): 94-101.
8.  Macintyre, Geoff, et al. “Copy number signatures and mutational
    processes in ovarian carcinoma.” ___Nature Genetics___ 50.9 (2018): 1262.
9.  Tan, Vincent YF, and Cédric Févotte. “Automatic relevance
    determination in nonnegative matrix factorization with the/spl
    beta/-divergence.” ___IEEE Transactions on Pattern Analysis and Machine Intelligence___ 35.7 (2012): 1592-1605.
10. Bergstrom EN, Huang MN, Mahto U, Barnes M, Stratton MR, Rozen SG,
    Alexandrov LB: SigProfilerMatrixGenerator: a tool for visualizing
    and exploring patterns of small mutational events. ___BMC Genomics___
    2019, 20:685
    <https://bmcgenomics.biomedcentral.com/articles/10.1186/s12864-019-6041-2>

## `r emoji("books")` Book structure {-}

+ Part 1 (Background and Prerequisite) describes the basic concepts of mutational signature and how to install/load **sigminer**. 
+ Part 2 (Workflows) introduces how to prepare your input data and run mutational signature analysis for different mutation data types (SBS, DBS, INDEL, Genome rearrangement, CNV) with different methods provided by **sigminer**.
+ Part 3 (Miscellaneous topics) describes useful utilities including builtin datasets, SBS signature conversion.

## `r emoji("sparkling_heart")` Want to help? {-}


The book’s source code is hosted on GitHub, at <https://github.com/ShixiangWang/sigminer-book>. Any feedback on the book is very welcome. Feel free to [open an issue](https://github.com/ShixiangWang/sigminer-book/issues/new) on GitHub or send me a pull request if you notice typos or other issues (I'm not a native English speaker ;) ).

## `r emoji("ant")` Bug report or feature request {-}

If you find any bugs or want to have a new feature, please [file an issue](https://github.com/ShixiangWang/sigminer/issues/new).


## `r emoji("cactus")` Acknowlegment {-}

I built this book website by imitating [*Biomedical knowledge mining using GOSemSim and clusterProfiler*](https://github.com/YuLab-SMU/biomedical-knowledge-mining-book) and reusing its configurations, I would like to thank [Guangchuang Yu](https://github.com/GuangchuangYu) here.
