<!DOCTYPE html>
<html lang="en">
<head>
<meta http-equiv="Content-Type" content="text/html; charset=UTF-8">
<meta charset="utf-8">
<meta name="viewport" content="width=device-width, initial-scale=1, shrink-to-fit=no">
<title>Chapter 4 Other signature types | Extract, Analyze and Visualize Mutational Signatures with Sigminer</title>
<meta name="author" content="Shixiang Wang, PhD (Sun Yat-sen University Cancer Center)">
<meta name="author" content="Xue-Song Liu, PhD (ShanghaiTech University)">
<meta name="description" content="4.1 Copy number signature (Wang et al)  4.1.1 Read data The input requires absolute copy number profile with following information: Segment chromosome. Segment start. Segment end. Absolute copy...">
<meta name="generator" content="bookdown 0.28 with bs4_book()">
<meta property="og:title" content="Chapter 4 Other signature types | Extract, Analyze and Visualize Mutational Signatures with Sigminer">
<meta property="og:type" content="book">
<meta property="og:description" content="4.1 Copy number signature (Wang et al)  4.1.1 Read data The input requires absolute copy number profile with following information: Segment chromosome. Segment start. Segment end. Absolute copy...">
<meta name="twitter:card" content="summary">
<meta name="twitter:title" content="Chapter 4 Other signature types | Extract, Analyze and Visualize Mutational Signatures with Sigminer">
<meta name="twitter:description" content="4.1 Copy number signature (Wang et al)  4.1.1 Read data The input requires absolute copy number profile with following information: Segment chromosome. Segment start. Segment end. Absolute copy...">
<!-- JS --><script src="https://cdnjs.cloudflare.com/ajax/libs/clipboard.js/2.0.6/clipboard.min.js" integrity="sha256-inc5kl9MA1hkeYUt+EC3BhlIgyp/2jDIyBLS6k3UxPI=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/fuse.js/6.4.6/fuse.js" integrity="sha512-zv6Ywkjyktsohkbp9bb45V6tEMoWhzFzXis+LrMehmJZZSys19Yxf1dopHx7WzIKxr5tK2dVcYmaCk2uqdjF4A==" crossorigin="anonymous"></script><script src="https://kit.fontawesome.com/6ecbd6c532.js" crossorigin="anonymous"></script><script src="libs/jquery-3.6.0/jquery-3.6.0.min.js"></script><meta name="viewport" content="width=device-width, initial-scale=1, shrink-to-fit=no">
<link href="libs/bootstrap-4.6.0/bootstrap.min.css" rel="stylesheet">
<script src="libs/bootstrap-4.6.0/bootstrap.bundle.min.js"></script><script src="libs/bs3compat-0.4.0/transition.js"></script><script src="libs/bs3compat-0.4.0/tabs.js"></script><script src="libs/bs3compat-0.4.0/bs3compat.js"></script><link href="libs/bs4_book-1.0.0/bs4_book.css" rel="stylesheet">
<script src="libs/bs4_book-1.0.0/bs4_book.js"></script><script src="libs/htmlwidgets-1.5.4/htmlwidgets.js"></script><link href="libs/datatables-css-0.0.0/datatables-crosstalk.css" rel="stylesheet">
<script src="libs/datatables-binding-0.24/datatables.js"></script><link href="libs/dt-core-1.11.3/css/jquery.dataTables.min.css" rel="stylesheet">
<link href="libs/dt-core-1.11.3/css/jquery.dataTables.extra.css" rel="stylesheet">
<script src="libs/dt-core-1.11.3/js/jquery.dataTables.min.js"></script><link href="libs/crosstalk-1.2.0/css/crosstalk.min.css" rel="stylesheet">
<script src="libs/crosstalk-1.2.0/js/crosstalk.min.js"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/autocomplete.js/0.38.0/autocomplete.jquery.min.js" integrity="sha512-GU9ayf+66Xx2TmpxqJpliWbT5PiGYxpaG8rfnBEk1LL8l1KGkRShhngwdXK1UgqhAzWpZHSiYPc09/NwDQIGyg==" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/mark.js/8.11.1/mark.min.js" integrity="sha512-5CYOlHXGh6QpOFA/TeTylKLWfB3ftPsde7AnmhuitiTX4K5SqCLBeKro6sPS8ilsz1Q4NRx3v8Ko2IBiszzdww==" crossorigin="anonymous"></script><!-- CSS --><style type="text/css">
    
    div.hanging-indent{margin-left: 1.5em; text-indent: -1.5em;}
  </style>
<style type="text/css">
    /* Used with Pandoc 2.11+ new --citeproc when CSL is used */
    div.csl-bib-body { }
    div.csl-entry {
      clear: both;
        }
    .hanging div.csl-entry {
      margin-left:2em;
      text-indent:-2em;
    }
    div.csl-left-margin {
      min-width:2em;
      float:left;
    }
    div.csl-right-inline {
      margin-left:2em;
      padding-left:1em;
    }
    div.csl-indent {
      margin-left: 2em;
    }
  </style>
</head>
<body data-spy="scroll" data-target="#toc">

<div class="container-fluid">
<div class="row">
  <header class="col-sm-12 col-lg-3 sidebar sidebar-book"><a class="sr-only sr-only-focusable" href="#content">Skip to main content</a>

    <div class="d-flex align-items-start justify-content-between">
      <h1>
        <a href="index.html" title="">Extract, Analyze and Visualize Mutational Signatures with Sigminer</a>
      </h1>
      <button class="btn btn-outline-primary d-lg-none ml-2 mt-1" type="button" data-toggle="collapse" data-target="#main-nav" aria-expanded="true" aria-controls="main-nav"><i class="fas fa-bars"></i><span class="sr-only">Show table of contents</span></button>
    </div>

    <div id="main-nav" class="collapse-lg">
      <form role="search">
        <input id="search" class="form-control" type="search" placeholder="Search" aria-label="Search">
</form>

      <nav aria-label="Table of contents"><h2>Table of contents</h2>
        <ul class="book-toc list-unstyled">
<li><a class="" href="index.html">📖 Introduction</a></li>
<li class="book-part">Part I: Background and Prerequisite</li>
<li><a class="" href="mutsig-intro.html"><span class="header-section-number">1</span> Mutational signatures</a></li>
<li><a class="" href="prerequisite.html"><span class="header-section-number">2</span> Package prerequisite and installation</a></li>
<li class="book-part">Part II: Workflows</li>
<li><a class="" href="basic-workflow.html"><span class="header-section-number">3</span> Mutational signature analysis basics</a></li>
<li><a class="active" href="analysis-supps.html"><span class="header-section-number">4</span> Other signature types</a></li>
<li><a class="" href="target-vis.html"><span class="header-section-number">5</span> Target visualization</a></li>
<li class="book-part">Part III: Miscellaneous topics</li>
<li><a class="" href="universal-analysis.html"><span class="header-section-number">6</span> Universal analysis</a></li>
<li><a class="" href="subtype-prediction.html"><span class="header-section-number">7</span> Subtype prediction</a></li>
<li><a class="" href="sigflow.html"><span class="header-section-number">8</span> Sigflow pipeline</a></li>
<li><a class="" href="datasets.html"><span class="header-section-number">9</span> Datasets</a></li>
<li><a class="" href="convert.html"><span class="header-section-number">10</span> SBS signature conversion</a></li>
<li class="book-part">Appendix</li>
<li><a class="" href="references.html">References</a></li>
</ul>

        <div class="book-extra">
          
        </div>
      </nav>
</div>
  </header><main class="col-sm-12 col-md-9 col-lg-7" id="content"><div id="analysis-supps" class="section level1" number="4">
<h1>
<span class="header-section-number">4</span> Other signature types<a class="anchor" aria-label="anchor" href="#analysis-supps"><i class="fas fa-link"></i></a>
</h1>
<div id="copy-number-signature-wang-et-al" class="section level2" number="4.1">
<h2>
<span class="header-section-number">4.1</span> Copy number signature (<em>Wang</em> et al)<a class="anchor" aria-label="anchor" href="#copy-number-signature-wang-et-al"><i class="fas fa-link"></i></a>
</h2>
<div id="read-data" class="section level3" number="4.1.1">
<h3>
<span class="header-section-number">4.1.1</span> Read data<a class="anchor" aria-label="anchor" href="#read-data"><i class="fas fa-link"></i></a>
</h3>
<p>The input requires absolute copy number profile with following information:</p>
<ul>
<li>Segment chromosome.</li>
<li>Segment start.</li>
<li>Segment end.</li>
<li>
<strong>Absolute copy number value for this segment</strong>: must be integer.</li>
<li>Sample ID.</li>
</ul>
<p>The input data can be result from any software which provides information above.</p>
<p>Useful softwares are listed below:</p>
<ul>
<li>
<a href="https://software.broadinstitute.org/cancer/cga/absolute">ABSOLUTE</a>.</li>
<li><a href="https://cran.r-project.org/web/packages/sequenza/index.html">Sequenza</a></li>
<li><a href="https://github.com/mskcc/facets">FACETS</a></li>
<li>
<a href="http://penncnv.openbioinformatics.org/en/latest/">PennCNV</a> &amp; <a href="https://www.crick.ac.uk/research/labs/peter-van-loo/software">ASCAT</a>
</li>
<li><a href="https://github.com/etal/cnvkit">CNVkit</a></li>
</ul>
<p>Useful analysis reference:</p>
<ul>
<li><a href="https://xsliulab.github.io/PC_CNA_signature/">Prostate Cancer Variation Signature Analysis Report</a></li>
</ul>
<p>The import work is done by <code><a href="https://rdrr.io/pkg/sigminer/man/read_copynumber.html">read_copynumber()</a></code>, which supports <code>data.frame</code> or file, and even result directory from <strong>ABSOLUTE</strong>.</p>
<p>Option <code>sigminer.sex</code> is used to control the processing of sex. If you don’t care the sex chromosomes (i.e. X and Y),
you can ignore this setting after removing the X/Y segments, otherwise the <code>summary</code> in the result <code>cn</code> and tally process may be biased.</p>
<div class="sourceCode" id="cb48"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co">## Default is "female"</span></span>
<span><span class="co">## You can ignore the setting if all samples are females</span></span>
<span><span class="co">## But we recommend you set it</span></span>
<span><span class="fu"><a href="https://rdrr.io/r/base/options.html">options</a></span><span class="op">(</span>sigminer.sex <span class="op">=</span> <span class="st">"male"</span><span class="op">)</span></span>
<span><span class="co">## For cohort contains both males and females,</span></span>
<span><span class="co">## set a data.frame with two columns, i.e.</span></span>
<span><span class="co">## options(sigminer.sex = sex_df),</span></span>
<span><span class="co">## which</span></span>
<span><span class="co">## sex_df = data.frame(sample = c("sample1", "sample2",</span></span>
<span><span class="co">##                     sex = "female", "male"))</span></span>
<span><span class="co"># Load toy dataset of absolute copynumber profile</span></span>
<span><span class="fu"><a href="https://rdrr.io/r/base/load.html">load</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/system.file.html">system.file</a></span><span class="op">(</span><span class="st">"extdata"</span>, <span class="st">"toy_segTab.RData"</span>,</span>
<span>  package <span class="op">=</span> <span class="st">"sigminer"</span>, mustWork <span class="op">=</span> <span class="cn">TRUE</span></span>
<span><span class="op">)</span><span class="op">)</span></span>
<span><span class="va">cn</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/pkg/sigminer/man/read_copynumber.html">read_copynumber</a></span><span class="op">(</span><span class="va">segTabs</span>,</span>
<span>  seg_cols <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html">c</a></span><span class="op">(</span><span class="st">"chromosome"</span>, <span class="st">"start"</span>, <span class="st">"end"</span>, <span class="st">"segVal"</span><span class="op">)</span>,</span>
<span>  genome_build <span class="op">=</span> <span class="st">"hg19"</span>, complement <span class="op">=</span> <span class="cn">FALSE</span>, verbose <span class="op">=</span> <span class="cn">TRUE</span></span>
<span><span class="op">)</span></span>
<span><span class="co">## ℹ [2021-12-04 23:51:19]: Started.</span></span>
<span><span class="co">## ℹ [2021-12-04 23:51:19]: Genome build  : hg19.</span></span>
<span><span class="co">## ℹ [2021-12-04 23:51:19]: Genome measure: called.</span></span>
<span><span class="co">## ✓ [2021-12-04 23:51:19]: Chromosome size database for build obtained.</span></span>
<span><span class="co">## ℹ [2021-12-04 23:51:19]: Reading input.</span></span>
<span><span class="co">## ✓ [2021-12-04 23:51:19]: A data frame as input detected.</span></span>
<span><span class="co">## ✓ [2021-12-04 23:51:19]: Column names checked.</span></span>
<span><span class="co">## ✓ [2021-12-04 23:51:19]: Column order set.</span></span>
<span><span class="co">## ✓ [2021-12-04 23:51:19]: Chromosomes unified.</span></span>
<span><span class="co">## ✓ [2021-12-04 23:51:19]: Data imported.</span></span>
<span><span class="co">## ℹ [2021-12-04 23:51:19]: Segments info:</span></span>
<span><span class="co">## ℹ [2021-12-04 23:51:19]:     Keep - 467</span></span>
<span><span class="co">## ℹ [2021-12-04 23:51:19]:     Drop - 0</span></span>
<span><span class="co">## ✓ [2021-12-04 23:51:19]: Segments sorted.</span></span>
<span><span class="co">## ℹ [2021-12-04 23:51:19]: Joining adjacent segments with same copy number value. Be patient...</span></span>
<span><span class="co">## ✓ [2021-12-04 23:51:19]: 400 segments left after joining.</span></span>
<span><span class="co">## ✓ [2021-12-04 23:51:19]: Segmental table cleaned.</span></span>
<span><span class="co">## ℹ [2021-12-04 23:51:19]: Annotating.</span></span>
<span><span class="co">## ✓ [2021-12-04 23:51:19]: Annotation done.</span></span>
<span><span class="co">## ℹ [2021-12-04 23:51:19]: Summarizing per sample.</span></span>
<span><span class="co">## ✓ [2021-12-04 23:51:19]: Summarized.</span></span>
<span><span class="co">## ℹ [2021-12-04 23:51:19]: Generating CopyNumber object.</span></span>
<span><span class="co">## ✓ [2021-12-04 23:51:19]: Generated.</span></span>
<span><span class="co">## ℹ [2021-12-04 23:51:19]: Validating object.</span></span>
<span><span class="co">## ✓ [2021-12-04 23:51:19]: Done.</span></span>
<span><span class="co">## ℹ [2021-12-04 23:51:19]: 0.167 secs elapsed.</span></span>
<span><span class="va">cn</span></span>
<span><span class="co">## An object of class CopyNumber </span></span>
<span><span class="co">## =============================</span></span>
<span><span class="co">##                           sample n_of_seg n_of_cnv n_of_amp n_of_del n_of_vchr</span></span>
<span><span class="co">##  1: TCGA-DF-A2KN-01A-11D-A17U-01       33        6        5        1         4</span></span>
<span><span class="co">##  2: TCGA-19-2621-01B-01D-0911-01       33        8        5        3         5</span></span>
<span><span class="co">##  3: TCGA-B6-A0X5-01A-21D-A107-01       28        8        4        4         2</span></span>
<span><span class="co">##  4: TCGA-A8-A07S-01A-11D-A036-01       38       11        2        9         4</span></span>
<span><span class="co">##  5: TCGA-26-6174-01A-21D-1842-01       43       13        8        5         8</span></span>
<span><span class="co">##  6: TCGA-CV-7432-01A-11D-2128-01       40       16        7        9         9</span></span>
<span><span class="co">##  7: TCGA-06-0644-01A-02D-0310-01       46       19        5       14         8</span></span>
<span><span class="co">##  8: TCGA-A5-A0G2-01A-11D-A042-01       39       21        5       16        10</span></span>
<span><span class="co">##  9: TCGA-99-7458-01A-11D-2035-01       48       26       10       16        13</span></span>
<span><span class="co">## 10: TCGA-05-4417-01A-22D-1854-01       52       37       33        4        17</span></span>
<span><span class="co">##     cna_burden</span></span>
<span><span class="co">##  1:      0.000</span></span>
<span><span class="co">##  2:      0.099</span></span>
<span><span class="co">##  3:      0.087</span></span>
<span><span class="co">##  4:      0.112</span></span>
<span><span class="co">##  5:      0.119</span></span>
<span><span class="co">##  6:      0.198</span></span>
<span><span class="co">##  7:      0.165</span></span>
<span><span class="co">##  8:      0.393</span></span>
<span><span class="co">##  9:      0.318</span></span>
<span><span class="co">## 10:      0.654</span></span></code></pre></div>
<blockquote>
<p>Currently, you can refer to <code>extract_facets_cnv()</code> and <code>extract_seqz_cnv()</code> in <a href="https://github.com/ShixiangWang/prad_signature/blob/master/analysis/src/99-functions.R" class="uri">https://github.com/ShixiangWang/prad_signature/blob/master/analysis/src/99-functions.R</a> to
see how to get tidy data from a result directory of FACETS or Sequenza.
## Tally Components</p>
</blockquote>
<p>Currently, there are two methods for generating sample-by-component matrix.</p>
<p>Option <code>sigminer.copynumber.max</code> is used to control the processing of max copy number values. Run <code><a href="https://rdrr.io/pkg/sigminer/man/sig_tally.html">?sig_tally</a></code> to see more.</p>
<div class="sourceCode" id="cb49"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co">## Even you set max_copynumber = 20 in read_copynumber(),</span></span>
<span><span class="co">## the segmental copy number may be greater than 20</span></span>
<span><span class="co">## because for male samples, the X/Y segmental copy number</span></span>
<span><span class="co">## values will be doubled in tally process.</span></span>
<span><span class="co">## This setting will make copy number values of all segments</span></span>
<span><span class="co">## not greater than 20.</span></span>
<span><span class="fu"><a href="https://rdrr.io/r/base/options.html">options</a></span><span class="op">(</span>sigminer.copynumber.max <span class="op">=</span> <span class="fl">20</span><span class="op">)</span></span>
<span><span class="co"># Load copy number object</span></span>
<span><span class="fu"><a href="https://rdrr.io/r/base/load.html">load</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/system.file.html">system.file</a></span><span class="op">(</span><span class="st">"extdata"</span>, <span class="st">"toy_copynumber.RData"</span>,</span>
<span>  package <span class="op">=</span> <span class="st">"sigminer"</span>, mustWork <span class="op">=</span> <span class="cn">TRUE</span></span>
<span><span class="op">)</span><span class="op">)</span></span>
<span><span class="co"># Use method designed by Wang, Shixiang et al.</span></span>
<span><span class="va">cn_tally_W</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/pkg/sigminer/man/sig_tally.html">sig_tally</a></span><span class="op">(</span><span class="va">cn</span>, method <span class="op">=</span> <span class="st">"W"</span><span class="op">)</span></span></code></pre></div>
<blockquote>
<p>You can set <code>options(sigminer.sex = "male", sigminer.copynumber.max = 20)</code> at the top of your code
to avoid setting them in two places.</p>
<p><strong>Of note</strong>, the <code>sigminer.copynumber.max</code> option only has effect on <code><a href="https://rdrr.io/pkg/sigminer/man/sig_tally.html">sig_tally()</a></code> with method “W”,
the <code>sigminer.sex</code> option has effects on <code><a href="https://rdrr.io/pkg/sigminer/man/read_copynumber.html">read_copynumber()</a></code> and <code><a href="https://rdrr.io/pkg/sigminer/man/sig_tally.html">sig_tally()</a></code> with method “W”.</p>
</blockquote>
<p>This step return a <code>list</code> containing information about copy number features, components and matrix for NMF etc.</p>
</div>
<div id="extract-signatures" class="section level3" number="4.1.2">
<h3>
<span class="header-section-number">4.1.2</span> Extract signatures<a class="anchor" aria-label="anchor" href="#extract-signatures"><i class="fas fa-link"></i></a>
</h3>
<p>When you get the matrix, you can just do the signature extraction as SBS etc. signatures. So here we won’t talk much.</p>
<div class="sourceCode" id="cb50"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">cn_tally_W</span><span class="op">$</span><span class="va">nmf_matrix</span><span class="op">[</span><span class="fl">1</span><span class="op">:</span><span class="fl">5</span>, <span class="fl">1</span><span class="op">:</span><span class="fl">5</span><span class="op">]</span></span>
<span><span class="co">##                              BP10MB[0] BP10MB[1] BP10MB[2] BP10MB[3] BP10MB[4]</span></span>
<span><span class="co">## TCGA-05-4417-01A-22D-1854-01       275        20         5         0         0</span></span>
<span><span class="co">## TCGA-06-0644-01A-02D-0310-01       289         5         4         0         1</span></span>
<span><span class="co">## TCGA-19-2621-01B-01D-0911-01       294         2         3         1         0</span></span>
<span><span class="co">## TCGA-26-6174-01A-21D-1842-01       288         4         7         1         0</span></span>
<span><span class="co">## TCGA-99-7458-01A-11D-2035-01       284         9         5         1         1</span></span></code></pre></div>
<div class="sourceCode" id="cb51"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co"># library(NMF)</span></span>
<span><span class="va">sig_w</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/pkg/sigminer/man/sig_extract.html">sig_extract</a></span><span class="op">(</span><span class="va">cn_tally_W</span><span class="op">$</span><span class="va">nmf_matrix</span>, n_sig <span class="op">=</span> <span class="fl">2</span><span class="op">)</span></span></code></pre></div>
</div>
</div>
<div id="allele-specific-copy-number-signature-steele-et-al" class="section level2" number="4.2">
<h2>
<span class="header-section-number">4.2</span> Allele specific copy number signature (<em>Steele</em> et al)<a class="anchor" aria-label="anchor" href="#allele-specific-copy-number-signature-steele-et-al"><i class="fas fa-link"></i></a>
</h2>
<p>Previously described steps can be also applied to allele specific copy number
profile with <em>Steele</em> et al approach. There are two catalog methods provided by
<em>Steele</em> et al: 40 catalogs and 48 catalogs. In current stage, 48 catalog is
recommended as ~20 reference signatures have been built on ~10,000 TCGA tumors.
reference example is given below:</p>
<div class="sourceCode" id="cb52"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb52-1"><a href="analysis-supps.html#cb52-1" aria-hidden="true" tabindex="-1"></a><span class="co"># Generate example data</span></span>
<span id="cb52-2"><a href="analysis-supps.html#cb52-2" aria-hidden="true" tabindex="-1"></a><span class="fu">load</span>(<span class="fu">system.file</span>(<span class="st">"extdata"</span>, <span class="st">"toy_segTab.RData"</span>,</span>
<span id="cb52-3"><a href="analysis-supps.html#cb52-3" aria-hidden="true" tabindex="-1"></a>  <span class="at">package =</span> <span class="st">"sigminer"</span>, <span class="at">mustWork =</span> <span class="cn">TRUE</span></span>
<span id="cb52-4"><a href="analysis-supps.html#cb52-4" aria-hidden="true" tabindex="-1"></a>))</span>
<span id="cb52-5"><a href="analysis-supps.html#cb52-5" aria-hidden="true" tabindex="-1"></a><span class="fu">set.seed</span>(<span class="dv">1234</span>)</span>
<span id="cb52-6"><a href="analysis-supps.html#cb52-6" aria-hidden="true" tabindex="-1"></a><span class="co"># Make sure minor_cn is provided</span></span>
<span id="cb52-7"><a href="analysis-supps.html#cb52-7" aria-hidden="true" tabindex="-1"></a>segTabs<span class="sc">$</span>minor_cn <span class="ot">&lt;-</span> <span class="fu">sample</span>(<span class="fu">c</span>(<span class="dv">0</span>, <span class="dv">1</span>), <span class="at">size =</span> <span class="fu">nrow</span>(segTabs), <span class="at">replace =</span> <span class="cn">TRUE</span>)</span>
<span id="cb52-8"><a href="analysis-supps.html#cb52-8" aria-hidden="true" tabindex="-1"></a>cn2 <span class="ot">&lt;-</span> <span class="fu">read_copynumber</span>(segTabs,</span>
<span id="cb52-9"><a href="analysis-supps.html#cb52-9" aria-hidden="true" tabindex="-1"></a>  <span class="at">seg_cols =</span> <span class="fu">c</span>(<span class="st">"chromosome"</span>, <span class="st">"start"</span>, <span class="st">"end"</span>, <span class="st">"segVal"</span>),</span>
<span id="cb52-10"><a href="analysis-supps.html#cb52-10" aria-hidden="true" tabindex="-1"></a>  <span class="at">genome_measure =</span> <span class="st">"wg"</span>, <span class="at">complement =</span> <span class="cn">TRUE</span>, <span class="at">add_loh =</span> <span class="cn">TRUE</span></span>
<span id="cb52-11"><a href="analysis-supps.html#cb52-11" aria-hidden="true" tabindex="-1"></a>)</span>
<span id="cb52-12"><a href="analysis-supps.html#cb52-12" aria-hidden="true" tabindex="-1"></a><span class="do">## ℹ [2022-08-29 11:59:13]: Started.</span></span>
<span id="cb52-13"><a href="analysis-supps.html#cb52-13" aria-hidden="true" tabindex="-1"></a><span class="do">## ℹ [2022-08-29 11:59:13]: Genome build  : hg19.</span></span>
<span id="cb52-14"><a href="analysis-supps.html#cb52-14" aria-hidden="true" tabindex="-1"></a><span class="do">## ℹ [2022-08-29 11:59:13]: Genome measure: wg.</span></span>
<span id="cb52-15"><a href="analysis-supps.html#cb52-15" aria-hidden="true" tabindex="-1"></a><span class="do">## ℹ [2022-08-29 11:59:13]: When add_loh is TRUE, use_all is forced to TRUE.</span></span>
<span id="cb52-16"><a href="analysis-supps.html#cb52-16" aria-hidden="true" tabindex="-1"></a><span class="do">## Please drop columns you don't want to keep before reading.</span></span>
<span id="cb52-17"><a href="analysis-supps.html#cb52-17" aria-hidden="true" tabindex="-1"></a><span class="do">## ✔ [2022-08-29 11:59:13]: Chromosome size database for build obtained.</span></span>
<span id="cb52-18"><a href="analysis-supps.html#cb52-18" aria-hidden="true" tabindex="-1"></a><span class="do">## ℹ [2022-08-29 11:59:13]: Reading input.</span></span>
<span id="cb52-19"><a href="analysis-supps.html#cb52-19" aria-hidden="true" tabindex="-1"></a><span class="do">## ✔ [2022-08-29 11:59:13]: A data frame as input detected.</span></span>
<span id="cb52-20"><a href="analysis-supps.html#cb52-20" aria-hidden="true" tabindex="-1"></a><span class="do">## ✔ [2022-08-29 11:59:13]: Column names checked.</span></span>
<span id="cb52-21"><a href="analysis-supps.html#cb52-21" aria-hidden="true" tabindex="-1"></a><span class="do">## ✔ [2022-08-29 11:59:13]: Column order set.</span></span>
<span id="cb52-22"><a href="analysis-supps.html#cb52-22" aria-hidden="true" tabindex="-1"></a><span class="do">## ✔ [2022-08-29 11:59:13]: Chromosomes unified.</span></span>
<span id="cb52-23"><a href="analysis-supps.html#cb52-23" aria-hidden="true" tabindex="-1"></a><span class="do">## ✔ [2022-08-29 11:59:13]: Value 2 (normal copy) filled to uncalled chromosomes.</span></span>
<span id="cb52-24"><a href="analysis-supps.html#cb52-24" aria-hidden="true" tabindex="-1"></a><span class="do">## ✔ [2022-08-29 11:59:13]: Data imported.</span></span>
<span id="cb52-25"><a href="analysis-supps.html#cb52-25" aria-hidden="true" tabindex="-1"></a><span class="do">## ℹ [2022-08-29 11:59:13]: Segments info:</span></span>
<span id="cb52-26"><a href="analysis-supps.html#cb52-26" aria-hidden="true" tabindex="-1"></a><span class="do">## ℹ [2022-08-29 11:59:13]:     Keep - 477</span></span>
<span id="cb52-27"><a href="analysis-supps.html#cb52-27" aria-hidden="true" tabindex="-1"></a><span class="do">## ℹ [2022-08-29 11:59:13]:     Drop - 0</span></span>
<span id="cb52-28"><a href="analysis-supps.html#cb52-28" aria-hidden="true" tabindex="-1"></a><span class="do">## ✔ [2022-08-29 11:59:13]: Segments sorted.</span></span>
<span id="cb52-29"><a href="analysis-supps.html#cb52-29" aria-hidden="true" tabindex="-1"></a><span class="do">## ℹ [2022-08-29 11:59:13]: Adding LOH labels...</span></span>
<span id="cb52-30"><a href="analysis-supps.html#cb52-30" aria-hidden="true" tabindex="-1"></a><span class="do">## ℹ [2022-08-29 11:59:13]: Joining adjacent segments with same copy number value. Be patient...</span></span>
<span id="cb52-31"><a href="analysis-supps.html#cb52-31" aria-hidden="true" tabindex="-1"></a><span class="do">## ✔ [2022-08-29 11:59:13]: 410 segments left after joining.</span></span>
<span id="cb52-32"><a href="analysis-supps.html#cb52-32" aria-hidden="true" tabindex="-1"></a><span class="do">## ✔ [2022-08-29 11:59:13]: Segmental table cleaned.</span></span>
<span id="cb52-33"><a href="analysis-supps.html#cb52-33" aria-hidden="true" tabindex="-1"></a><span class="do">## ℹ [2022-08-29 11:59:13]: Annotating.</span></span>
<span id="cb52-34"><a href="analysis-supps.html#cb52-34" aria-hidden="true" tabindex="-1"></a><span class="do">## ✔ [2022-08-29 11:59:13]: Annotation done.</span></span>
<span id="cb52-35"><a href="analysis-supps.html#cb52-35" aria-hidden="true" tabindex="-1"></a><span class="do">## ℹ [2022-08-29 11:59:13]: Summarizing per sample.</span></span>
<span id="cb52-36"><a href="analysis-supps.html#cb52-36" aria-hidden="true" tabindex="-1"></a><span class="do">## ✔ [2022-08-29 11:59:13]: Summarized.</span></span>
<span id="cb52-37"><a href="analysis-supps.html#cb52-37" aria-hidden="true" tabindex="-1"></a><span class="do">## ℹ [2022-08-29 11:59:13]: Generating CopyNumber object.</span></span>
<span id="cb52-38"><a href="analysis-supps.html#cb52-38" aria-hidden="true" tabindex="-1"></a><span class="do">## ✔ [2022-08-29 11:59:13]: Generated.</span></span>
<span id="cb52-39"><a href="analysis-supps.html#cb52-39" aria-hidden="true" tabindex="-1"></a><span class="do">## ℹ [2022-08-29 11:59:13]: Validating object.</span></span>
<span id="cb52-40"><a href="analysis-supps.html#cb52-40" aria-hidden="true" tabindex="-1"></a><span class="do">## ✔ [2022-08-29 11:59:13]: Done.</span></span>
<span id="cb52-41"><a href="analysis-supps.html#cb52-41" aria-hidden="true" tabindex="-1"></a><span class="do">## ℹ [2022-08-29 11:59:13]: 0.454 secs elapsed.</span></span>
<span id="cb52-42"><a href="analysis-supps.html#cb52-42" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb52-43"><a href="analysis-supps.html#cb52-43" aria-hidden="true" tabindex="-1"></a><span class="co"># Use tally method "S" (Steele et al.)</span></span>
<span id="cb52-44"><a href="analysis-supps.html#cb52-44" aria-hidden="true" tabindex="-1"></a>tally_s2 <span class="ot">&lt;-</span> <span class="fu">sig_tally</span>(cn2, <span class="at">method =</span> <span class="st">"S"</span>)</span>
<span id="cb52-45"><a href="analysis-supps.html#cb52-45" aria-hidden="true" tabindex="-1"></a><span class="do">## ℹ [2022-08-29 11:59:13]: Started.</span></span>
<span id="cb52-46"><a href="analysis-supps.html#cb52-46" aria-hidden="true" tabindex="-1"></a><span class="do">## ℹ [2022-08-29 11:59:13]: When you use method 'S', please make sure you have set 'join_adj_seg' to FALSE and 'add_loh' to TRUE in 'read_copynumber() in the previous step!</span></span>
<span id="cb52-47"><a href="analysis-supps.html#cb52-47" aria-hidden="true" tabindex="-1"></a><span class="do">## ✔ [2022-08-29 11:59:13]: Matrix generated.</span></span>
<span id="cb52-48"><a href="analysis-supps.html#cb52-48" aria-hidden="true" tabindex="-1"></a><span class="do">## ℹ [2022-08-29 11:59:13]: 0.05 secs elapsed.</span></span>
<span id="cb52-49"><a href="analysis-supps.html#cb52-49" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb52-50"><a href="analysis-supps.html#cb52-50" aria-hidden="true" tabindex="-1"></a>cn_sig2 <span class="ot">&lt;-</span> <span class="fu">sig_extract</span>(tally_s2<span class="sc">$</span>all_matrices<span class="sc">$</span>CN_48, <span class="at">n_sig =</span> <span class="dv">2</span>)</span>
<span id="cb52-51"><a href="analysis-supps.html#cb52-51" aria-hidden="true" tabindex="-1"></a><span class="do">## NMF algorithm: 'brunet'</span></span>
<span id="cb52-52"><a href="analysis-supps.html#cb52-52" aria-hidden="true" tabindex="-1"></a><span class="do">## Multiple runs: 10</span></span>
<span id="cb52-53"><a href="analysis-supps.html#cb52-53" aria-hidden="true" tabindex="-1"></a><span class="do">## Mode: sequential [foreach:doParallelMC]</span></span>
<span id="cb52-54"><a href="analysis-supps.html#cb52-54" aria-hidden="true" tabindex="-1"></a><span class="do">## </span></span>
<span id="cb52-55"><a href="analysis-supps.html#cb52-55" aria-hidden="true" tabindex="-1"></a>Runs<span class="sc">:</span> <span class="er">|</span>                                                        </span>
<span id="cb52-56"><a href="analysis-supps.html#cb52-56" aria-hidden="true" tabindex="-1"></a>Runs<span class="sc">:</span> <span class="er">|</span>                                                  <span class="er">|</span>   <span class="dv">0</span>%</span>
<span id="cb52-57"><a href="analysis-supps.html#cb52-57" aria-hidden="true" tabindex="-1"></a>Runs<span class="sc">:</span> <span class="er">|</span>                                                        </span>
<span id="cb52-58"><a href="analysis-supps.html#cb52-58" aria-hidden="true" tabindex="-1"></a>Runs<span class="sc">:</span> <span class="er">|=====</span>                                             <span class="er">|</span>   <span class="dv">9</span>%</span>
<span id="cb52-59"><a href="analysis-supps.html#cb52-59" aria-hidden="true" tabindex="-1"></a>Runs<span class="sc">:</span> <span class="er">|</span>                                                        </span>
<span id="cb52-60"><a href="analysis-supps.html#cb52-60" aria-hidden="true" tabindex="-1"></a>Runs<span class="sc">:</span> <span class="er">|=========</span>                                         <span class="er">|</span>  <span class="dv">18</span>%</span>
<span id="cb52-61"><a href="analysis-supps.html#cb52-61" aria-hidden="true" tabindex="-1"></a>Runs<span class="sc">:</span> <span class="er">|</span>                                                        </span>
<span id="cb52-62"><a href="analysis-supps.html#cb52-62" aria-hidden="true" tabindex="-1"></a>Runs<span class="sc">:</span> <span class="er">|==============</span>                                    <span class="er">|</span>  <span class="dv">27</span>%</span>
<span id="cb52-63"><a href="analysis-supps.html#cb52-63" aria-hidden="true" tabindex="-1"></a>Runs<span class="sc">:</span> <span class="er">|</span>                                                        </span>
<span id="cb52-64"><a href="analysis-supps.html#cb52-64" aria-hidden="true" tabindex="-1"></a>Runs<span class="sc">:</span> <span class="er">|==================</span>                                <span class="er">|</span>  <span class="dv">36</span>%</span>
<span id="cb52-65"><a href="analysis-supps.html#cb52-65" aria-hidden="true" tabindex="-1"></a>Runs<span class="sc">:</span> <span class="er">|</span>                                                        </span>
<span id="cb52-66"><a href="analysis-supps.html#cb52-66" aria-hidden="true" tabindex="-1"></a>Runs<span class="sc">:</span> <span class="er">|=======================</span>                           <span class="er">|</span>  <span class="dv">45</span>%</span>
<span id="cb52-67"><a href="analysis-supps.html#cb52-67" aria-hidden="true" tabindex="-1"></a>Runs<span class="sc">:</span> <span class="er">|</span>                                                        </span>
<span id="cb52-68"><a href="analysis-supps.html#cb52-68" aria-hidden="true" tabindex="-1"></a>Runs<span class="sc">:</span> <span class="er">|===========================</span>                       <span class="er">|</span>  <span class="dv">55</span>%</span>
<span id="cb52-69"><a href="analysis-supps.html#cb52-69" aria-hidden="true" tabindex="-1"></a>Runs<span class="sc">:</span> <span class="er">|</span>                                                        </span>
<span id="cb52-70"><a href="analysis-supps.html#cb52-70" aria-hidden="true" tabindex="-1"></a>Runs<span class="sc">:</span> <span class="er">|================================</span>                  <span class="er">|</span>  <span class="dv">64</span>%</span>
<span id="cb52-71"><a href="analysis-supps.html#cb52-71" aria-hidden="true" tabindex="-1"></a>Runs<span class="sc">:</span> <span class="er">|</span>                                                        </span>
<span id="cb52-72"><a href="analysis-supps.html#cb52-72" aria-hidden="true" tabindex="-1"></a>Runs<span class="sc">:</span> <span class="er">|====================================</span>              <span class="er">|</span>  <span class="dv">73</span>%</span>
<span id="cb52-73"><a href="analysis-supps.html#cb52-73" aria-hidden="true" tabindex="-1"></a>Runs<span class="sc">:</span> <span class="er">|</span>                                                        </span>
<span id="cb52-74"><a href="analysis-supps.html#cb52-74" aria-hidden="true" tabindex="-1"></a>Runs<span class="sc">:</span> <span class="er">|=========================================</span>         <span class="er">|</span>  <span class="dv">82</span>%</span>
<span id="cb52-75"><a href="analysis-supps.html#cb52-75" aria-hidden="true" tabindex="-1"></a>Runs<span class="sc">:</span> <span class="er">|</span>                                                        </span>
<span id="cb52-76"><a href="analysis-supps.html#cb52-76" aria-hidden="true" tabindex="-1"></a>Runs<span class="sc">:</span> <span class="er">|=============================================</span>     <span class="er">|</span>  <span class="dv">91</span>%</span>
<span id="cb52-77"><a href="analysis-supps.html#cb52-77" aria-hidden="true" tabindex="-1"></a>Runs<span class="sc">:</span> <span class="er">|</span>                                                        </span>
<span id="cb52-78"><a href="analysis-supps.html#cb52-78" aria-hidden="true" tabindex="-1"></a>Runs<span class="sc">:</span> <span class="er">|==================================================|</span> <span class="dv">100</span>%</span>
<span id="cb52-79"><a href="analysis-supps.html#cb52-79" aria-hidden="true" tabindex="-1"></a><span class="do">## System time:</span></span>
<span id="cb52-80"><a href="analysis-supps.html#cb52-80" aria-hidden="true" tabindex="-1"></a><span class="do">##    user  system elapsed </span></span>
<span id="cb52-81"><a href="analysis-supps.html#cb52-81" aria-hidden="true" tabindex="-1"></a><span class="do">##   9.021   0.119   9.163</span></span>
<span id="cb52-82"><a href="analysis-supps.html#cb52-82" aria-hidden="true" tabindex="-1"></a><span class="fu">get_sig_similarity</span>(cn_sig2, <span class="at">sig_db =</span> <span class="st">"CNS_TCGA"</span>)</span>
<span id="cb52-83"><a href="analysis-supps.html#cb52-83" aria-hidden="true" tabindex="-1"></a><span class="do">## -Comparing against COSMIC signatures</span></span>
<span id="cb52-84"><a href="analysis-supps.html#cb52-84" aria-hidden="true" tabindex="-1"></a><span class="do">## ------------------------------------</span></span>
<span id="cb52-85"><a href="analysis-supps.html#cb52-85" aria-hidden="true" tabindex="-1"></a><span class="do">## --Found Sig1 most similar to CN1</span></span>
<span id="cb52-86"><a href="analysis-supps.html#cb52-86" aria-hidden="true" tabindex="-1"></a><span class="do">##    Aetiology: See https://cancer.sanger.ac.uk/signatures/cn/ [similarity: 0.728]</span></span>
<span id="cb52-87"><a href="analysis-supps.html#cb52-87" aria-hidden="true" tabindex="-1"></a><span class="do">## --Found Sig2 most similar to CN2</span></span>
<span id="cb52-88"><a href="analysis-supps.html#cb52-88" aria-hidden="true" tabindex="-1"></a><span class="do">##    Aetiology: See https://cancer.sanger.ac.uk/signatures/cn/ [similarity: 0.668]</span></span>
<span id="cb52-89"><a href="analysis-supps.html#cb52-89" aria-hidden="true" tabindex="-1"></a><span class="do">## ------------------------------------</span></span>
<span id="cb52-90"><a href="analysis-supps.html#cb52-90" aria-hidden="true" tabindex="-1"></a><span class="do">## Return result invisiblely.</span></span></code></pre></div>
</div>
<div id="rearrangement-signature" class="section level2" number="4.3">
<h2>
<span class="header-section-number">4.3</span> Rearrangement signature<a class="anchor" aria-label="anchor" href="#rearrangement-signature"><i class="fas fa-link"></i></a>
</h2>
<p>Similarity, once you know how to generate matrix for genome rearrangement signature,
you can easily apply the signature extraction or signature fitting.</p>
<div class="sourceCode" id="cb53"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">sv</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/readRDS.html">readRDS</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/system.file.html">system.file</a></span><span class="op">(</span><span class="st">"extdata"</span>, <span class="st">"toy_sv.rds"</span>, package <span class="op">=</span> <span class="st">"sigminer"</span>, mustWork <span class="op">=</span> <span class="cn">TRUE</span><span class="op">)</span><span class="op">)</span></span>
<span><span class="va">rs</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/pkg/sigminer/man/read_sv_as_rs.html">read_sv_as_rs</a></span><span class="op">(</span><span class="va">sv</span><span class="op">)</span></span>
<span><span class="co">## succesfully read RS!</span></span>
<span><span class="co"># svclass is optional</span></span>
<span><span class="va">rs2</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/pkg/sigminer/man/read_sv_as_rs.html">read_sv_as_rs</a></span><span class="op">(</span><span class="va">sv</span><span class="op">[</span>, <span class="fu"><a href="https://rdrr.io/r/base/sets.html">setdiff</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/colnames.html">colnames</a></span><span class="op">(</span><span class="va">sv</span><span class="op">)</span>, <span class="st">"svclass"</span><span class="op">)</span><span class="op">]</span><span class="op">)</span></span>
<span><span class="co">## succesfully read RS!</span></span>
<span><span class="fu"><a href="https://rdrr.io/r/base/identical.html">identical</a></span><span class="op">(</span><span class="va">rs</span>, <span class="va">rs2</span><span class="op">)</span></span>
<span><span class="co">## [1] TRUE</span></span>
<span></span>
<span><span class="va">tally_rs</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/pkg/sigminer/man/sig_tally.html">sig_tally</a></span><span class="op">(</span><span class="va">rs</span><span class="op">)</span></span>
<span><span class="co">## ℹ [2022-08-29 11:59:24]: Started.</span></span>
<span><span class="co">## ✔ [2022-08-29 11:59:24]: Successfully get RS list!</span></span>
<span><span class="co">## </span></span>
<span><span class="co">## Attaching package: 'purrr'</span></span>
<span><span class="co">## The following objects are masked from 'package:foreach':</span></span>
<span><span class="co">## </span></span>
<span><span class="co">##     accumulate, when</span></span>
<span><span class="co">## The following object is masked from 'package:XVector':</span></span>
<span><span class="co">## </span></span>
<span><span class="co">##     compact</span></span>
<span><span class="co">## The following object is masked from 'package:GenomicRanges':</span></span>
<span><span class="co">## </span></span>
<span><span class="co">##     reduce</span></span>
<span><span class="co">## The following object is masked from 'package:IRanges':</span></span>
<span><span class="co">## </span></span>
<span><span class="co">##     reduce</span></span>
<span><span class="co">## [1] "Getting clustered info..."</span></span>
<span><span class="co">## pcf finished for chromosome arm 1p </span></span>
<span><span class="co">## pcf finished for chromosome arm 10p </span></span>
<span><span class="co">## pcf finished for chromosome arm 10q </span></span>
<span><span class="co">## pcf finished for chromosome arm 11p </span></span>
<span><span class="co">## pcf finished for chromosome arm 11q </span></span>
<span><span class="co">## pcf finished for chromosome arm 12p </span></span>
<span><span class="co">## pcf finished for chromosome arm 12q </span></span>
<span><span class="co">## pcf finished for chromosome arm 13q </span></span>
<span><span class="co">## pcf finished for chromosome arm 14q </span></span>
<span><span class="co">## pcf finished for chromosome arm 16q </span></span>
<span><span class="co">## pcf finished for chromosome arm 17p </span></span>
<span><span class="co">## pcf finished for chromosome arm 17q </span></span>
<span><span class="co">## pcf finished for chromosome arm 18p </span></span>
<span><span class="co">## pcf finished for chromosome arm 18q </span></span>
<span><span class="co">## pcf finished for chromosome arm 19p </span></span>
<span><span class="co">## pcf finished for chromosome arm 19q </span></span>
<span><span class="co">## pcf finished for chromosome arm 2p </span></span>
<span><span class="co">## pcf finished for chromosome arm 2q </span></span>
<span><span class="co">## pcf finished for chromosome arm 20p </span></span>
<span><span class="co">## pcf finished for chromosome arm 20q </span></span>
<span><span class="co">## pcf finished for chromosome arm 22q </span></span>
<span><span class="co">## pcf finished for chromosome arm 3p </span></span>
<span><span class="co">## pcf finished for chromosome arm 3q </span></span>
<span><span class="co">## pcf finished for chromosome arm 4p </span></span>
<span><span class="co">## pcf finished for chromosome arm 4q </span></span>
<span><span class="co">## pcf finished for chromosome arm 5p </span></span>
<span><span class="co">## pcf finished for chromosome arm 5q </span></span>
<span><span class="co">## pcf finished for chromosome arm 6p </span></span>
<span><span class="co">## pcf finished for chromosome arm 6q </span></span>
<span><span class="co">## pcf finished for chromosome arm 7p </span></span>
<span><span class="co">## pcf finished for chromosome arm 7q </span></span>
<span><span class="co">## pcf finished for chromosome arm 8p </span></span>
<span><span class="co">## pcf finished for chromosome arm 8q </span></span>
<span><span class="co">## pcf finished for chromosome arm 9p </span></span>
<span><span class="co">## pcf finished for chromosome arm 9q </span></span>
<span><span class="co">## pcf finished for chromosome arm Xp </span></span>
<span><span class="co">## pcf finished for chromosome arm Xq </span></span>
<span><span class="co">## pcf finished for chromosome arm 21q </span></span>
<span><span class="co">## pcf finished for chromosome arm 1p </span></span>
<span><span class="co">## pcf finished for chromosome arm 1q </span></span>
<span><span class="co">## pcf finished for chromosome arm 10p </span></span>
<span><span class="co">## pcf finished for chromosome arm 10q </span></span>
<span><span class="co">## pcf finished for chromosome arm 11q </span></span>
<span><span class="co">## pcf finished for chromosome arm 12p </span></span>
<span><span class="co">## pcf finished for chromosome arm 12q </span></span>
<span><span class="co">## pcf finished for chromosome arm 13q </span></span>
<span><span class="co">## pcf finished for chromosome arm 14q </span></span>
<span><span class="co">## pcf finished for chromosome arm 15q </span></span>
<span><span class="co">## pcf finished for chromosome arm 16p </span></span>
<span><span class="co">## pcf finished for chromosome arm 16q </span></span>
<span><span class="co">## pcf finished for chromosome arm 17p </span></span>
<span><span class="co">## pcf finished for chromosome arm 17q </span></span>
<span><span class="co">## pcf finished for chromosome arm 18q </span></span>
<span><span class="co">## pcf finished for chromosome arm 19q </span></span>
<span><span class="co">## pcf finished for chromosome arm 2p </span></span>
<span><span class="co">## pcf finished for chromosome arm 2q </span></span>
<span><span class="co">## pcf finished for chromosome arm 20q </span></span>
<span><span class="co">## pcf finished for chromosome arm 21p </span></span>
<span><span class="co">## pcf finished for chromosome arm 21q </span></span>
<span><span class="co">## pcf finished for chromosome arm 22q </span></span>
<span><span class="co">## pcf finished for chromosome arm 3p </span></span>
<span><span class="co">## pcf finished for chromosome arm 3q </span></span>
<span><span class="co">## pcf finished for chromosome arm 4p </span></span>
<span><span class="co">## pcf finished for chromosome arm 4q </span></span>
<span><span class="co">## pcf finished for chromosome arm 5p </span></span>
<span><span class="co">## pcf finished for chromosome arm 5q </span></span>
<span><span class="co">## pcf finished for chromosome arm 6p </span></span>
<span><span class="co">## pcf finished for chromosome arm 7q </span></span>
<span><span class="co">## pcf finished for chromosome arm 8p </span></span>
<span><span class="co">## pcf finished for chromosome arm 8q </span></span>
<span><span class="co">## pcf finished for chromosome arm 9p </span></span>
<span><span class="co">## pcf finished for chromosome arm 9q </span></span>
<span><span class="co">## pcf finished for chromosome arm Xp </span></span>
<span><span class="co">## pcf finished for chromosome arm Xq </span></span>
<span><span class="co">## pcf finished for chromosome arm 1p </span></span>
<span><span class="co">## pcf finished for chromosome arm 1q </span></span>
<span><span class="co">## pcf finished for chromosome arm 10q </span></span>
<span><span class="co">## pcf finished for chromosome arm 11p </span></span>
<span><span class="co">## pcf finished for chromosome arm 11q </span></span>
<span><span class="co">## pcf finished for chromosome arm 12p </span></span>
<span><span class="co">## pcf finished for chromosome arm 13q </span></span>
<span><span class="co">## pcf finished for chromosome arm 14q </span></span>
<span><span class="co">## pcf finished for chromosome arm 15q </span></span>
<span><span class="co">## pcf finished for chromosome arm 16p </span></span>
<span><span class="co">## pcf finished for chromosome arm 17p </span></span>
<span><span class="co">## pcf finished for chromosome arm 17q </span></span>
<span><span class="co">## pcf finished for chromosome arm 18p </span></span>
<span><span class="co">## pcf finished for chromosome arm 18q </span></span>
<span><span class="co">## pcf finished for chromosome arm 19p </span></span>
<span><span class="co">## pcf finished for chromosome arm 19q </span></span>
<span><span class="co">## pcf finished for chromosome arm 2p </span></span>
<span><span class="co">## pcf finished for chromosome arm 2q </span></span>
<span><span class="co">## pcf finished for chromosome arm 20p </span></span>
<span><span class="co">## pcf finished for chromosome arm 20q </span></span>
<span><span class="co">## pcf finished for chromosome arm 21q </span></span>
<span><span class="co">## pcf finished for chromosome arm 3p </span></span>
<span><span class="co">## pcf finished for chromosome arm 3q </span></span>
<span><span class="co">## pcf finished for chromosome arm 4p </span></span>
<span><span class="co">## pcf finished for chromosome arm 5q </span></span>
<span><span class="co">## pcf finished for chromosome arm 6p </span></span>
<span><span class="co">## pcf finished for chromosome arm 6q </span></span>
<span><span class="co">## pcf finished for chromosome arm 7q </span></span>
<span><span class="co">## pcf finished for chromosome arm 8p </span></span>
<span><span class="co">## pcf finished for chromosome arm 8q </span></span>
<span><span class="co">## pcf finished for chromosome arm 9p </span></span>
<span><span class="co">## pcf finished for chromosome arm 9q </span></span>
<span><span class="co">## pcf finished for chromosome arm Xq </span></span>
<span><span class="co">## pcf finished for chromosome arm 22q </span></span>
<span><span class="co">## pcf finished for chromosome arm 1p </span></span>
<span><span class="co">## pcf finished for chromosome arm 1q </span></span>
<span><span class="co">## pcf finished for chromosome arm 10q </span></span>
<span><span class="co">## pcf finished for chromosome arm 11p </span></span>
<span><span class="co">## pcf finished for chromosome arm 11q </span></span>
<span><span class="co">## pcf finished for chromosome arm 12p </span></span>
<span><span class="co">## pcf finished for chromosome arm 12q </span></span>
<span><span class="co">## pcf finished for chromosome arm 14q </span></span>
<span><span class="co">## pcf finished for chromosome arm 15q </span></span>
<span><span class="co">## pcf finished for chromosome arm 16p </span></span>
<span><span class="co">## pcf finished for chromosome arm 16q </span></span>
<span><span class="co">## pcf finished for chromosome arm 17p </span></span>
<span><span class="co">## pcf finished for chromosome arm 17q </span></span>
<span><span class="co">## pcf finished for chromosome arm 18q </span></span>
<span><span class="co">## pcf finished for chromosome arm 2p </span></span>
<span><span class="co">## pcf finished for chromosome arm 2q </span></span>
<span><span class="co">## pcf finished for chromosome arm 20p </span></span>
<span><span class="co">## pcf finished for chromosome arm 20q </span></span>
<span><span class="co">## pcf finished for chromosome arm 21q </span></span>
<span><span class="co">## pcf finished for chromosome arm 3p </span></span>
<span><span class="co">## pcf finished for chromosome arm 3q </span></span>
<span><span class="co">## pcf finished for chromosome arm 4p </span></span>
<span><span class="co">## pcf finished for chromosome arm 4q </span></span>
<span><span class="co">## pcf finished for chromosome arm 5p </span></span>
<span><span class="co">## pcf finished for chromosome arm 5q </span></span>
<span><span class="co">## pcf finished for chromosome arm 6p </span></span>
<span><span class="co">## pcf finished for chromosome arm 6q </span></span>
<span><span class="co">## pcf finished for chromosome arm 7p </span></span>
<span><span class="co">## pcf finished for chromosome arm 7q </span></span>
<span><span class="co">## pcf finished for chromosome arm 8p </span></span>
<span><span class="co">## pcf finished for chromosome arm 8q </span></span>
<span><span class="co">## pcf finished for chromosome arm 9p </span></span>
<span><span class="co">## pcf finished for chromosome arm 22q </span></span>
<span><span class="co">## pcf finished for chromosome arm Xq </span></span>
<span><span class="co">## pcf finished for chromosome arm 13q </span></span>
<span><span class="co">## pcf finished for chromosome arm 1p </span></span>
<span><span class="co">## pcf finished for chromosome arm 1q </span></span>
<span><span class="co">## pcf finished for chromosome arm 10p </span></span>
<span><span class="co">## pcf finished for chromosome arm 11p </span></span>
<span><span class="co">## pcf finished for chromosome arm 11q </span></span>
<span><span class="co">## pcf finished for chromosome arm 12p </span></span>
<span><span class="co">## pcf finished for chromosome arm 13q </span></span>
<span><span class="co">## pcf finished for chromosome arm 14q </span></span>
<span><span class="co">## pcf finished for chromosome arm 15q </span></span>
<span><span class="co">## pcf finished for chromosome arm 16q </span></span>
<span><span class="co">## pcf finished for chromosome arm 17p </span></span>
<span><span class="co">## pcf finished for chromosome arm 17q </span></span>
<span><span class="co">## pcf finished for chromosome arm 2p </span></span>
<span><span class="co">## pcf finished for chromosome arm 2q </span></span>
<span><span class="co">## pcf finished for chromosome arm 20q </span></span>
<span><span class="co">## pcf finished for chromosome arm 4q </span></span>
<span><span class="co">## pcf finished for chromosome arm 5p </span></span>
<span><span class="co">## pcf finished for chromosome arm 5q </span></span>
<span><span class="co">## pcf finished for chromosome arm 6p </span></span>
<span><span class="co">## pcf finished for chromosome arm 6q </span></span>
<span><span class="co">## pcf finished for chromosome arm 7p </span></span>
<span><span class="co">## pcf finished for chromosome arm 8p </span></span>
<span><span class="co">## pcf finished for chromosome arm 8q </span></span>
<span><span class="co">## pcf finished for chromosome arm 9q </span></span>
<span><span class="co">## pcf finished for chromosome arm Xp </span></span>
<span><span class="co">## pcf finished for chromosome arm Xq </span></span>
<span><span class="co">## pcf finished for chromosome arm 21p </span></span>
<span><span class="co">## pcf finished for chromosome arm 22q </span></span>
<span><span class="co">## pcf finished for chromosome arm 1p </span></span>
<span><span class="co">## pcf finished for chromosome arm 1q </span></span>
<span><span class="co">## pcf finished for chromosome arm 10p </span></span>
<span><span class="co">## pcf finished for chromosome arm 10q </span></span>
<span><span class="co">## pcf finished for chromosome arm 11p </span></span>
<span><span class="co">## pcf finished for chromosome arm 11q </span></span>
<span><span class="co">## pcf finished for chromosome arm 12p </span></span>
<span><span class="co">## pcf finished for chromosome arm 12q </span></span>
<span><span class="co">## pcf finished for chromosome arm 13q </span></span>
<span><span class="co">## pcf finished for chromosome arm 14q </span></span>
<span><span class="co">## pcf finished for chromosome arm 15q </span></span>
<span><span class="co">## pcf finished for chromosome arm 16p </span></span>
<span><span class="co">## pcf finished for chromosome arm 16q </span></span>
<span><span class="co">## pcf finished for chromosome arm 17p </span></span>
<span><span class="co">## pcf finished for chromosome arm 17q </span></span>
<span><span class="co">## pcf finished for chromosome arm 18p </span></span>
<span><span class="co">## pcf finished for chromosome arm 18q </span></span>
<span><span class="co">## pcf finished for chromosome arm 19p </span></span>
<span><span class="co">## pcf finished for chromosome arm 2p </span></span>
<span><span class="co">## pcf finished for chromosome arm 2q </span></span>
<span><span class="co">## pcf finished for chromosome arm 22q </span></span>
<span><span class="co">## pcf finished for chromosome arm 3p </span></span>
<span><span class="co">## pcf finished for chromosome arm 3q </span></span>
<span><span class="co">## pcf finished for chromosome arm 4p </span></span>
<span><span class="co">## pcf finished for chromosome arm 4q </span></span>
<span><span class="co">## pcf finished for chromosome arm 5p </span></span>
<span><span class="co">## pcf finished for chromosome arm 5q </span></span>
<span><span class="co">## pcf finished for chromosome arm 6p </span></span>
<span><span class="co">## pcf finished for chromosome arm 6q </span></span>
<span><span class="co">## pcf finished for chromosome arm 7p </span></span>
<span><span class="co">## pcf finished for chromosome arm 7q </span></span>
<span><span class="co">## pcf finished for chromosome arm 8p </span></span>
<span><span class="co">## pcf finished for chromosome arm 8q </span></span>
<span><span class="co">## pcf finished for chromosome arm 9p </span></span>
<span><span class="co">## pcf finished for chromosome arm 9q </span></span>
<span><span class="co">## pcf finished for chromosome arm Xp </span></span>
<span><span class="co">## pcf finished for chromosome arm Xq </span></span>
<span><span class="co">## pcf finished for chromosome arm 20q </span></span>
<span><span class="co">## pcf finished for chromosome arm 1p </span></span>
<span><span class="co">## pcf finished for chromosome arm 1q </span></span>
<span><span class="co">## pcf finished for chromosome arm 10p </span></span>
<span><span class="co">## pcf finished for chromosome arm 10q </span></span>
<span><span class="co">## pcf finished for chromosome arm 11p </span></span>
<span><span class="co">## pcf finished for chromosome arm 11q </span></span>
<span><span class="co">## pcf finished for chromosome arm 12p </span></span>
<span><span class="co">## pcf finished for chromosome arm 12q </span></span>
<span><span class="co">## pcf finished for chromosome arm 13q </span></span>
<span><span class="co">## pcf finished for chromosome arm 14q </span></span>
<span><span class="co">## pcf finished for chromosome arm 15q </span></span>
<span><span class="co">## pcf finished for chromosome arm 16p </span></span>
<span><span class="co">## pcf finished for chromosome arm 16q </span></span>
<span><span class="co">## pcf finished for chromosome arm 17p </span></span>
<span><span class="co">## pcf finished for chromosome arm 17q </span></span>
<span><span class="co">## pcf finished for chromosome arm 18q </span></span>
<span><span class="co">## pcf finished for chromosome arm 19p </span></span>
<span><span class="co">## pcf finished for chromosome arm 19q </span></span>
<span><span class="co">## pcf finished for chromosome arm 2p </span></span>
<span><span class="co">## pcf finished for chromosome arm 2q </span></span>
<span><span class="co">## pcf finished for chromosome arm 20p </span></span>
<span><span class="co">## pcf finished for chromosome arm 20q </span></span>
<span><span class="co">## pcf finished for chromosome arm 21q </span></span>
<span><span class="co">## pcf finished for chromosome arm 22q </span></span>
<span><span class="co">## pcf finished for chromosome arm 3p </span></span>
<span><span class="co">## pcf finished for chromosome arm 4p </span></span>
<span><span class="co">## pcf finished for chromosome arm 4q </span></span>
<span><span class="co">## pcf finished for chromosome arm 5p </span></span>
<span><span class="co">## pcf finished for chromosome arm 5q </span></span>
<span><span class="co">## pcf finished for chromosome arm 6p </span></span>
<span><span class="co">## pcf finished for chromosome arm 6q </span></span>
<span><span class="co">## pcf finished for chromosome arm 7p </span></span>
<span><span class="co">## pcf finished for chromosome arm 7q </span></span>
<span><span class="co">## pcf finished for chromosome arm 8p </span></span>
<span><span class="co">## pcf finished for chromosome arm 8q </span></span>
<span><span class="co">## pcf finished for chromosome arm 9p </span></span>
<span><span class="co">## pcf finished for chromosome arm 9q </span></span>
<span><span class="co">## pcf finished for chromosome arm Xp </span></span>
<span><span class="co">## pcf finished for chromosome arm Xq </span></span>
<span><span class="co">## pcf finished for chromosome arm Yq </span></span>
<span><span class="co">## pcf finished for chromosome arm 1p </span></span>
<span><span class="co">## pcf finished for chromosome arm 1q </span></span>
<span><span class="co">## pcf finished for chromosome arm 10p </span></span>
<span><span class="co">## pcf finished for chromosome arm 10q </span></span>
<span><span class="co">## pcf finished for chromosome arm 11p </span></span>
<span><span class="co">## pcf finished for chromosome arm 11q </span></span>
<span><span class="co">## pcf finished for chromosome arm 12p </span></span>
<span><span class="co">## pcf finished for chromosome arm 12q </span></span>
<span><span class="co">## pcf finished for chromosome arm 13q </span></span>
<span><span class="co">## pcf finished for chromosome arm 14q </span></span>
<span><span class="co">## pcf finished for chromosome arm 15q </span></span>
<span><span class="co">## pcf finished for chromosome arm 16p </span></span>
<span><span class="co">## pcf finished for chromosome arm 16q </span></span>
<span><span class="co">## pcf finished for chromosome arm 17p </span></span>
<span><span class="co">## pcf finished for chromosome arm 17q </span></span>
<span><span class="co">## pcf finished for chromosome arm 18p </span></span>
<span><span class="co">## pcf finished for chromosome arm 18q </span></span>
<span><span class="co">## pcf finished for chromosome arm 19p </span></span>
<span><span class="co">## pcf finished for chromosome arm 19q </span></span>
<span><span class="co">## pcf finished for chromosome arm 2p </span></span>
<span><span class="co">## pcf finished for chromosome arm 2q </span></span>
<span><span class="co">## pcf finished for chromosome arm 20p </span></span>
<span><span class="co">## pcf finished for chromosome arm 20q </span></span>
<span><span class="co">## pcf finished for chromosome arm 21q </span></span>
<span><span class="co">## pcf finished for chromosome arm 22q </span></span>
<span><span class="co">## pcf finished for chromosome arm 3p </span></span>
<span><span class="co">## pcf finished for chromosome arm 4p </span></span>
<span><span class="co">## pcf finished for chromosome arm 4q </span></span>
<span><span class="co">## pcf finished for chromosome arm 5p </span></span>
<span><span class="co">## pcf finished for chromosome arm 5q </span></span>
<span><span class="co">## pcf finished for chromosome arm 6p </span></span>
<span><span class="co">## pcf finished for chromosome arm 6q </span></span>
<span><span class="co">## pcf finished for chromosome arm 7p </span></span>
<span><span class="co">## pcf finished for chromosome arm 7q </span></span>
<span><span class="co">## pcf finished for chromosome arm 8p </span></span>
<span><span class="co">## pcf finished for chromosome arm 8q </span></span>
<span><span class="co">## pcf finished for chromosome arm 9p </span></span>
<span><span class="co">## pcf finished for chromosome arm 9q </span></span>
<span><span class="co">## pcf finished for chromosome arm Xp </span></span>
<span><span class="co">## pcf finished for chromosome arm Xq </span></span>
<span><span class="co">## pcf finished for chromosome arm 1p </span></span>
<span><span class="co">## pcf finished for chromosome arm 10q </span></span>
<span><span class="co">## pcf finished for chromosome arm 11q </span></span>
<span><span class="co">## pcf finished for chromosome arm 12p </span></span>
<span><span class="co">## pcf finished for chromosome arm 12q </span></span>
<span><span class="co">## pcf finished for chromosome arm 13q </span></span>
<span><span class="co">## pcf finished for chromosome arm 15q </span></span>
<span><span class="co">## pcf finished for chromosome arm 16q </span></span>
<span><span class="co">## pcf finished for chromosome arm 17p </span></span>
<span><span class="co">## pcf finished for chromosome arm 19p </span></span>
<span><span class="co">## pcf finished for chromosome arm 19q </span></span>
<span><span class="co">## pcf finished for chromosome arm 2p </span></span>
<span><span class="co">## pcf finished for chromosome arm 22q </span></span>
<span><span class="co">## pcf finished for chromosome arm 3q </span></span>
<span><span class="co">## pcf finished for chromosome arm 5p </span></span>
<span><span class="co">## pcf finished for chromosome arm 5q </span></span>
<span><span class="co">## pcf finished for chromosome arm 6q </span></span>
<span><span class="co">## pcf finished for chromosome arm 7p </span></span>
<span><span class="co">## pcf finished for chromosome arm 7q </span></span>
<span><span class="co">## pcf finished for chromosome arm 9p </span></span>
<span><span class="co">## pcf finished for chromosome arm 9q </span></span>
<span><span class="co">## pcf finished for chromosome arm Xq </span></span>
<span><span class="co">## pcf finished for chromosome arm 1p </span></span>
<span><span class="co">## pcf finished for chromosome arm 11q </span></span>
<span><span class="co">## pcf finished for chromosome arm 13q </span></span>
<span><span class="co">## pcf finished for chromosome arm 14q </span></span>
<span><span class="co">## pcf finished for chromosome arm 15q </span></span>
<span><span class="co">## pcf finished for chromosome arm 16p </span></span>
<span><span class="co">## pcf finished for chromosome arm 16q </span></span>
<span><span class="co">## pcf finished for chromosome arm 17p </span></span>
<span><span class="co">## pcf finished for chromosome arm 17q </span></span>
<span><span class="co">## pcf finished for chromosome arm 18p </span></span>
<span><span class="co">## pcf finished for chromosome arm 18q </span></span>
<span><span class="co">## pcf finished for chromosome arm 19p </span></span>
<span><span class="co">## pcf finished for chromosome arm 19q </span></span>
<span><span class="co">## pcf finished for chromosome arm 2p </span></span>
<span><span class="co">## pcf finished for chromosome arm 2q </span></span>
<span><span class="co">## pcf finished for chromosome arm 20p </span></span>
<span><span class="co">## pcf finished for chromosome arm 20q </span></span>
<span><span class="co">## pcf finished for chromosome arm 21p </span></span>
<span><span class="co">## pcf finished for chromosome arm 21q </span></span>
<span><span class="co">## pcf finished for chromosome arm 22q </span></span>
<span><span class="co">## pcf finished for chromosome arm 3p </span></span>
<span><span class="co">## pcf finished for chromosome arm 3q </span></span>
<span><span class="co">## pcf finished for chromosome arm 4p </span></span>
<span><span class="co">## pcf finished for chromosome arm 4q </span></span>
<span><span class="co">## pcf finished for chromosome arm 5p </span></span>
<span><span class="co">## pcf finished for chromosome arm 5q </span></span>
<span><span class="co">## pcf finished for chromosome arm 6p </span></span>
<span><span class="co">## pcf finished for chromosome arm 6q </span></span>
<span><span class="co">## pcf finished for chromosome arm 7p </span></span>
<span><span class="co">## pcf finished for chromosome arm 7q </span></span>
<span><span class="co">## pcf finished for chromosome arm 8p </span></span>
<span><span class="co">## pcf finished for chromosome arm 8q </span></span>
<span><span class="co">## pcf finished for chromosome arm 9p </span></span>
<span><span class="co">## pcf finished for chromosome arm 9q </span></span>
<span><span class="co">## pcf finished for chromosome arm Xp </span></span>
<span><span class="co">## pcf finished for chromosome arm Xq </span></span>
<span><span class="co">## pcf finished for chromosome arm Yp </span></span>
<span><span class="co">## pcf finished for chromosome arm Yq </span></span>
<span><span class="co">## [1] "Getting type of segment ..."</span></span>
<span><span class="co">## [1] "Getting distance of two rearrange segments ..."</span></span>
<span><span class="co">## ✔ [2022-08-29 11:59:28]: Successfully get RS features!</span></span>
<span><span class="co">## ✔ [2022-08-29 11:59:28]: Successfully get RS component!</span></span>
<span><span class="co">## ✔ [2022-08-29 11:59:28]: Successfully get RS matrix!</span></span>
<span><span class="co">## ℹ [2022-08-29 11:59:28]: 3.937 secs elapsed.</span></span>
<span><span class="fu"><a href="https://rdrr.io/r/utils/str.html">str</a></span><span class="op">(</span><span class="va">tally_rs</span><span class="op">)</span></span>
<span><span class="co">## List of 4</span></span>
<span><span class="co">##  $ features    :List of 3</span></span>
<span><span class="co">##   ..$ clustered:Classes 'data.table' and 'data.frame':   4220 obs. of  3 variables:</span></span>
<span><span class="co">##   .. ..$ sample: chr [1:4220] "PD26861a" "PD26861a" "PD26861a" "PD26861a" ...</span></span>
<span><span class="co">##   .. ..$ value : chr [1:4220] "non-clustered" "non-clustered" "non-clustered" "non-clustered" ...</span></span>
<span><span class="co">##   .. ..$ Index : int [1:4220] 1 2 3 4 5 6 7 8 9 10 ...</span></span>
<span><span class="co">##   .. ..- attr(*, ".internal.selfref")=&lt;externalptr&gt; </span></span>
<span><span class="co">##   ..$ type     :Classes 'data.table' and 'data.frame':   4220 obs. of  3 variables:</span></span>
<span><span class="co">##   .. ..$ sample: chr [1:4220] "PD26861a" "PD26861a" "PD26861a" "PD26861a" ...</span></span>
<span><span class="co">##   .. ..$ value : chr [1:4220] "inv" "inv" "del" "del" ...</span></span>
<span><span class="co">##   .. ..$ Index : int [1:4220] 1 2 3 4 5 6 7 8 9 10 ...</span></span>
<span><span class="co">##   .. ..- attr(*, ".internal.selfref")=&lt;externalptr&gt; </span></span>
<span><span class="co">##   ..$ size     :Classes 'data.table' and 'data.frame':   4220 obs. of  3 variables:</span></span>
<span><span class="co">##   .. ..$ sample: chr [1:4220] "PD26861a" "PD26861a" "PD26861a" "PD26861a" ...</span></span>
<span><span class="co">##   .. ..$ value : chr [1:4220] "&gt;10Mb" "1Mb-10Mb" "&gt;10Mb" "100Kb-1Mb" ...</span></span>
<span><span class="co">##   .. ..$ Index : int [1:4220] 1 2 3 4 5 6 7 8 9 10 ...</span></span>
<span><span class="co">##   .. ..- attr(*, ".internal.selfref")=&lt;externalptr&gt; </span></span>
<span><span class="co">##  $ components  :List of 3</span></span>
<span><span class="co">##   ..$ clustered:Classes 'data.table' and 'data.frame':   4220 obs. of  3 variables:</span></span>
<span><span class="co">##   .. ..$ sample     : chr [1:4220] "PD26861a" "PD26861a" "PD26861a" "PD26861a" ...</span></span>
<span><span class="co">##   .. ..$ Index      : int [1:4220] 1 2 3 4 5 6 7 8 9 10 ...</span></span>
<span><span class="co">##   .. ..$ C_clustered: Factor w/ 2 levels "clustered","non-clustered": 2 2 2 2 2 2 2 2 2 2 ...</span></span>
<span><span class="co">##   .. ..- attr(*, ".internal.selfref")=&lt;externalptr&gt; </span></span>
<span><span class="co">##   ..$ type     :Classes 'data.table' and 'data.frame':   4220 obs. of  3 variables:</span></span>
<span><span class="co">##   .. ..$ sample: chr [1:4220] "PD26861a" "PD26861a" "PD26861a" "PD26861a" ...</span></span>
<span><span class="co">##   .. ..$ Index : int [1:4220] 1 2 3 4 5 6 7 8 9 10 ...</span></span>
<span><span class="co">##   .. ..$ C_type: Factor w/ 4 levels "del","inv","tds",..: 2 2 1 1 2 2 1 1 2 3 ...</span></span>
<span><span class="co">##   .. ..- attr(*, ".internal.selfref")=&lt;externalptr&gt; </span></span>
<span><span class="co">##   ..$ size     :Classes 'data.table' and 'data.frame':   4220 obs. of  3 variables:</span></span>
<span><span class="co">##   .. ..$ sample: chr [1:4220] "PD26861a" "PD26861a" "PD26861a" "PD26861a" ...</span></span>
<span><span class="co">##   .. ..$ Index : int [1:4220] 1 2 3 4 5 6 7 8 9 10 ...</span></span>
<span><span class="co">##   .. ..$ C_size: Factor w/ 6 levels "&lt;1Kb","&gt;10Mb",..: 2 6 2 5 2 2 6 5 5 5 ...</span></span>
<span><span class="co">##   .. ..- attr(*, ".internal.selfref")=&lt;externalptr&gt; </span></span>
<span><span class="co">##  $ nmf_matrix  : int [1:10, 1:32] 13 3 2 0 1 3 2 11 5 1 ...</span></span>
<span><span class="co">##   ..- attr(*, "dimnames")=List of 2</span></span>
<span><span class="co">##   .. ..$ : chr [1:10] "PD26861a" "PD26862a" "PD26864a" "PD26865a" ...</span></span>
<span><span class="co">##   .. ..$ : chr [1:32] "clustered:del:&gt;10Mb" "clustered:del:1-10Kb" "clustered:del:10-100Kb" "clustered:del:100Kb-1Mb" ...</span></span>
<span><span class="co">##  $ all_matrices:List of 2</span></span>
<span><span class="co">##   ..$ RS_32: int [1:10, 1:32] 13 3 2 0 1 3 2 11 5 1 ...</span></span>
<span><span class="co">##   .. ..- attr(*, "dimnames")=List of 2</span></span>
<span><span class="co">##   .. .. ..$ : chr [1:10] "PD26861a" "PD26862a" "PD26864a" "PD26865a" ...</span></span>
<span><span class="co">##   .. .. ..$ : chr [1:32] "clustered:del:&gt;10Mb" "clustered:del:1-10Kb" "clustered:del:10-100Kb" "clustered:del:100Kb-1Mb" ...</span></span>
<span><span class="co">##   ..$ RS_38: int [1:10, 1:38] 0 0 0 0 0 0 0 0 0 0 ...</span></span>
<span><span class="co">##   .. ..- attr(*, "dimnames")=List of 2</span></span>
<span><span class="co">##   .. .. ..$ : chr [1:10] "PD26861a" "PD26862a" "PD26864a" "PD26865a" ...</span></span>
<span><span class="co">##   .. .. ..$ : chr [1:38] "clustered:del:&lt;1Kb" "clustered:del:&gt;10Mb" "clustered:del:1-10Kb" "clustered:del:10-100Kb" ...</span></span></code></pre></div>
</div>
</div>
  <div class="chapter-nav">
<div class="prev"><a href="basic-workflow.html"><span class="header-section-number">3</span> Mutational signature analysis basics</a></div>
<div class="next"><a href="target-vis.html"><span class="header-section-number">5</span> Target visualization</a></div>
</div></main><div class="col-md-3 col-lg-2 d-none d-md-block sidebar sidebar-chapter">
    <nav id="toc" data-toggle="toc" aria-label="On this page"><h2>On this page</h2>
      <ul class="nav navbar-nav">
<li><a class="nav-link" href="#analysis-supps"><span class="header-section-number">4</span> Other signature types</a></li>
<li>
<a class="nav-link" href="#copy-number-signature-wang-et-al"><span class="header-section-number">4.1</span> Copy number signature (Wang et al)</a><ul class="nav navbar-nav">
<li><a class="nav-link" href="#read-data"><span class="header-section-number">4.1.1</span> Read data</a></li>
<li><a class="nav-link" href="#extract-signatures"><span class="header-section-number">4.1.2</span> Extract signatures</a></li>
</ul>
</li>
<li><a class="nav-link" href="#allele-specific-copy-number-signature-steele-et-al"><span class="header-section-number">4.2</span> Allele specific copy number signature (Steele et al)</a></li>
<li><a class="nav-link" href="#rearrangement-signature"><span class="header-section-number">4.3</span> Rearrangement signature</a></li>
</ul>

      <div class="book-extra">
        <ul class="list-unstyled">
          
        </ul>
</div>
    </nav>
</div>

</div>
</div> <!-- .container -->

<footer class="bg-primary text-light mt-5"><div class="container"><div class="row">

  <div class="col-12 col-md-6 mt-3">
    <p>"<strong>Extract, Analyze and Visualize Mutational Signatures with Sigminer</strong>" was written by Shixiang Wang, PhD (Sun Yat-sen University Cancer Center), Xue-Song Liu, PhD (ShanghaiTech University). It was last built on 2022-08-29.</p>
  </div>

  <div class="col-12 col-md-6 mt-3">
    <p>This book was built by the <a class="text-light" href="https://bookdown.org">bookdown</a> R package.</p>
  </div>

</div></div>
</footer><!-- dynamically load mathjax for compatibility with self-contained --><script>
  (function () {
    var script = document.createElement("script");
    script.type = "text/javascript";
    var src = "true";
    if (src === "" || src === "true") src = "https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.9/latest.js?config=TeX-MML-AM_CHTML";
    if (location.protocol !== "file:")
      if (/^https?:/.test(src))
        src = src.replace(/^https?:/, '');
    script.src = src;
    document.getElementsByTagName("head")[0].appendChild(script);
  })();
</script><script type="text/x-mathjax-config">const popovers = document.querySelectorAll('a.footnote-ref[data-toggle="popover"]');
for (let popover of popovers) {
  const div = document.createElement('div');
  div.setAttribute('style', 'position: absolute; top: 0, left:0; width:0, height:0, overflow: hidden; visibility: hidden;');
  div.innerHTML = popover.getAttribute('data-content');

  var has_math = div.querySelector("span.math");
  if (has_math) {
    document.body.appendChild(div);
    MathJax.Hub.Queue(["Typeset", MathJax.Hub, div]);
    MathJax.Hub.Queue(function() {
      popover.setAttribute('data-content', div.innerHTML);
      document.body.removeChild(div);
    })
  }
}
</script>
</body>
</html>
