<!DOCTYPE html>
<html lang="en">
<head>
<meta http-equiv="Content-Type" content="text/html; charset=UTF-8">
<meta charset="utf-8">
<meta name="viewport" content="width=device-width, initial-scale=1, shrink-to-fit=no">
<title>Chapter 4 Analysis supplements | Extract, Analyze and Visualize Mutational Signatures with Sigminer</title>
<meta name="author" content="Shixiang Wang, PhD (Sun Yat-sen University Cancer Center)">
<meta name="author" content="Xue-Song Liu, PhD (ShanghaiTech University)">
<meta name="description" content="4.1 Copy number signature identification  4.1.1 Read data The input requires absolute copy number profile with following information: Segment chromosome. Segment start. Segment end. Absolute copy...">
<meta name="generator" content="bookdown 0.24 with bs4_book()">
<meta property="og:title" content="Chapter 4 Analysis supplements | Extract, Analyze and Visualize Mutational Signatures with Sigminer">
<meta property="og:type" content="book">
<meta property="og:description" content="4.1 Copy number signature identification  4.1.1 Read data The input requires absolute copy number profile with following information: Segment chromosome. Segment start. Segment end. Absolute copy...">
<meta name="twitter:card" content="summary">
<meta name="twitter:title" content="Chapter 4 Analysis supplements | Extract, Analyze and Visualize Mutational Signatures with Sigminer">
<meta name="twitter:description" content="4.1 Copy number signature identification  4.1.1 Read data The input requires absolute copy number profile with following information: Segment chromosome. Segment start. Segment end. Absolute copy...">
<!-- JS --><script src="https://cdnjs.cloudflare.com/ajax/libs/clipboard.js/2.0.6/clipboard.min.js" integrity="sha256-inc5kl9MA1hkeYUt+EC3BhlIgyp/2jDIyBLS6k3UxPI=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/fuse.js/6.4.6/fuse.js" integrity="sha512-zv6Ywkjyktsohkbp9bb45V6tEMoWhzFzXis+LrMehmJZZSys19Yxf1dopHx7WzIKxr5tK2dVcYmaCk2uqdjF4A==" crossorigin="anonymous"></script><script src="https://kit.fontawesome.com/6ecbd6c532.js" crossorigin="anonymous"></script><script src="libs/header-attrs-2.11/header-attrs.js"></script><script src="libs/jquery-3.6.0/jquery-3.6.0.min.js"></script><meta name="viewport" content="width=device-width, initial-scale=1, shrink-to-fit=no">
<link href="libs/bootstrap-4.6.0/bootstrap.min.css" rel="stylesheet">
<script src="libs/bootstrap-4.6.0/bootstrap.bundle.min.js"></script><script src="libs/bs3compat-0.3.1/transition.js"></script><script src="libs/bs3compat-0.3.1/tabs.js"></script><script src="libs/bs3compat-0.3.1/bs3compat.js"></script><link href="libs/bs4_book-1.0.0/bs4_book.css" rel="stylesheet">
<script src="libs/bs4_book-1.0.0/bs4_book.js"></script><script src="libs/htmlwidgets-1.5.4/htmlwidgets.js"></script><link href="libs/datatables-css-0.0.0/datatables-crosstalk.css" rel="stylesheet">
<script src="libs/datatables-binding-0.19/datatables.js"></script><link href="libs/dt-core-1.10.20/css/jquery.dataTables.min.css" rel="stylesheet">
<link href="libs/dt-core-1.10.20/css/jquery.dataTables.extra.css" rel="stylesheet">
<script src="libs/dt-core-1.10.20/js/jquery.dataTables.min.js"></script><link href="libs/crosstalk-1.1.1/css/crosstalk.css" rel="stylesheet">
<script src="libs/crosstalk-1.1.1/js/crosstalk.min.js"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/autocomplete.js/0.38.0/autocomplete.jquery.min.js" integrity="sha512-GU9ayf+66Xx2TmpxqJpliWbT5PiGYxpaG8rfnBEk1LL8l1KGkRShhngwdXK1UgqhAzWpZHSiYPc09/NwDQIGyg==" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/mark.js/8.11.1/mark.min.js" integrity="sha512-5CYOlHXGh6QpOFA/TeTylKLWfB3ftPsde7AnmhuitiTX4K5SqCLBeKro6sPS8ilsz1Q4NRx3v8Ko2IBiszzdww==" crossorigin="anonymous"></script><!-- CSS -->
</head>
<body data-spy="scroll" data-target="#toc">

<div class="container-fluid">
<div class="row">
  <header class="col-sm-12 col-lg-3 sidebar sidebar-book"><a class="sr-only sr-only-focusable" href="#content">Skip to main content</a>

    <div class="d-flex align-items-start justify-content-between">
      <h1>
        <a href="index.html" title="">Extract, Analyze and Visualize Mutational Signatures with Sigminer</a>
      </h1>
      <button class="btn btn-outline-primary d-lg-none ml-2 mt-1" type="button" data-toggle="collapse" data-target="#main-nav" aria-expanded="true" aria-controls="main-nav"><i class="fas fa-bars"></i><span class="sr-only">Show table of contents</span></button>
    </div>

    <div id="main-nav" class="collapse-lg">
      <form role="search">
        <input id="search" class="form-control" type="search" placeholder="Search" aria-label="Search">
</form>

      <nav aria-label="Table of contents"><h2>Table of contents</h2>
        <ul class="book-toc list-unstyled">
<li><a class="" href="index.html">üìñ Introduction</a></li>
<li class="book-part">Part I: Background and Prerequisite</li>
<li><a class="" href="mutsig-intro.html"><span class="header-section-number">1</span> Mutational signatures</a></li>
<li><a class="" href="prerequisite.html"><span class="header-section-number">2</span> Package prerequisite and installation</a></li>
<li class="book-part">Part II: Workflows</li>
<li><a class="" href="basic-workflow.html"><span class="header-section-number">3</span> Basic analysis workflow</a></li>
<li><a class="active" href="analysis-supps.html"><span class="header-section-number">4</span> Analysis supplements</a></li>
<li><a class="" href="target-vis.html"><span class="header-section-number">5</span> Target visualization</a></li>
<li class="book-part">Part III: Miscellaneous topics</li>
<li><a class="" href="universal-analysis.html"><span class="header-section-number">6</span> Universal analysis</a></li>
<li><a class="" href="subtype-prediction.html"><span class="header-section-number">7</span> Subtype prediction</a></li>
<li><a class="" href="sigflow.html"><span class="header-section-number">8</span> Sigflow pipeline</a></li>
<li><a class="" href="datasets.html"><span class="header-section-number">9</span> Datasets</a></li>
<li><a class="" href="convert.html"><span class="header-section-number">10</span> SBS signature conversion</a></li>
<li class="book-part">Appendix</li>
<li><a class="" href="references.html">References</a></li>
</ul>

        <div class="book-extra">
          
        </div>
      </nav>
</div>
  </header><main class="col-sm-12 col-md-9 col-lg-7" id="content"><div id="analysis-supps" class="section level1" number="4">
<h1>
<span class="header-section-number">4</span> Analysis supplements<a class="anchor" aria-label="anchor" href="#analysis-supps"><i class="fas fa-link"></i></a>
</h1>
<div id="copy-number-signature-identification" class="section level2" number="4.1">
<h2>
<span class="header-section-number">4.1</span> Copy number signature identification<a class="anchor" aria-label="anchor" href="#copy-number-signature-identification"><i class="fas fa-link"></i></a>
</h2>
<div id="read-data" class="section level3" number="4.1.1">
<h3>
<span class="header-section-number">4.1.1</span> Read data<a class="anchor" aria-label="anchor" href="#read-data"><i class="fas fa-link"></i></a>
</h3>
<p>The input requires absolute copy number profile with following information:</p>
<ul>
<li>Segment chromosome.</li>
<li>Segment start.</li>
<li>Segment end.</li>
<li>
<strong>Absolute copy number value for this segment</strong>: must be integer.</li>
<li>Sample ID.</li>
</ul>
<p>The input data can be result from any software which provides information above.</p>
<p>Useful softwares are listed below:</p>
<ul>
<li>
<a href="https://software.broadinstitute.org/cancer/cga/absolute">ABSOLUTE</a>.</li>
<li><a href="https://cran.r-project.org/web/packages/sequenza/index.html">Sequenza</a></li>
<li><a href="https://github.com/mskcc/facets">FACETS</a></li>
<li>
<a href="http://penncnv.openbioinformatics.org/en/latest/">PennCNV</a> &amp; <a href="https://www.crick.ac.uk/research/labs/peter-van-loo/software">ASCAT</a>
</li>
<li><a href="https://github.com/etal/cnvkit">CNVkit</a></li>
</ul>
<p>The import work is done by <code><a href="https://rdrr.io/pkg/sigminer/man/read_copynumber.html">read_copynumber()</a></code>, which supports <code>data.frame</code> or file, and even result directory from <strong>ABSOLUTE</strong>.</p>
<p>Option <code>sigminer.sex</code> is used to control the processing of sex. If you don‚Äôt care the sex chromosomes (i.e.¬†X and Y),
you can ignore this setting after removing the X/Y segments, otherwise the <code>summary</code> in the result <code>cn</code> and tally process may be biased.</p>
<div class="sourceCode" id="cb48"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="co">## Default is "female"</span>
<span class="co">## You can ignore the setting if all samples are females</span>
<span class="co">## But we recommend you set it</span>
<span class="fu"><a href="https://rdrr.io/r/base/options.html">options</a></span><span class="op">(</span>sigminer.sex <span class="op">=</span> <span class="st">"male"</span><span class="op">)</span>
<span class="co">## For cohort contains both males and females,</span>
<span class="co">## set a data.frame with two columns, i.e.</span>
<span class="co">## options(sigminer.sex = sex_df),</span>
<span class="co">## which</span>
<span class="co">## sex_df = data.frame(sample = c("sample1", "sample2",</span>
<span class="co">##                     sex = "female", "male"))</span>
<span class="co"># Load toy dataset of absolute copynumber profile</span>
<span class="fu"><a href="https://rdrr.io/r/base/load.html">load</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/system.file.html">system.file</a></span><span class="op">(</span><span class="st">"extdata"</span>, <span class="st">"toy_segTab.RData"</span>,
  package <span class="op">=</span> <span class="st">"sigminer"</span>, mustWork <span class="op">=</span> <span class="cn">TRUE</span>
<span class="op">)</span><span class="op">)</span>
<span class="va">cn</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/pkg/sigminer/man/read_copynumber.html">read_copynumber</a></span><span class="op">(</span><span class="va">segTabs</span>,
  seg_cols <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html">c</a></span><span class="op">(</span><span class="st">"chromosome"</span>, <span class="st">"start"</span>, <span class="st">"end"</span>, <span class="st">"segVal"</span><span class="op">)</span>,
  genome_build <span class="op">=</span> <span class="st">"hg19"</span>, complement <span class="op">=</span> <span class="cn">FALSE</span>, verbose <span class="op">=</span> <span class="cn">TRUE</span>
<span class="op">)</span>
<span class="co">## ‚Ñπ [2021-12-04 23:51:19]: Started.</span>
<span class="co">## ‚Ñπ [2021-12-04 23:51:19]: Genome build  : hg19.</span>
<span class="co">## ‚Ñπ [2021-12-04 23:51:19]: Genome measure: called.</span>
<span class="co">## ‚úì [2021-12-04 23:51:19]: Chromosome size database for build obtained.</span>
<span class="co">## ‚Ñπ [2021-12-04 23:51:19]: Reading input.</span>
<span class="co">## ‚úì [2021-12-04 23:51:19]: A data frame as input detected.</span>
<span class="co">## ‚úì [2021-12-04 23:51:19]: Column names checked.</span>
<span class="co">## ‚úì [2021-12-04 23:51:19]: Column order set.</span>
<span class="co">## ‚úì [2021-12-04 23:51:19]: Chromosomes unified.</span>
<span class="co">## ‚úì [2021-12-04 23:51:19]: Data imported.</span>
<span class="co">## ‚Ñπ [2021-12-04 23:51:19]: Segments info:</span>
<span class="co">## ‚Ñπ [2021-12-04 23:51:19]:     Keep - 467</span>
<span class="co">## ‚Ñπ [2021-12-04 23:51:19]:     Drop - 0</span>
<span class="co">## ‚úì [2021-12-04 23:51:19]: Segments sorted.</span>
<span class="co">## ‚Ñπ [2021-12-04 23:51:19]: Joining adjacent segments with same copy number value. Be patient...</span>
<span class="co">## ‚úì [2021-12-04 23:51:19]: 400 segments left after joining.</span>
<span class="co">## ‚úì [2021-12-04 23:51:19]: Segmental table cleaned.</span>
<span class="co">## ‚Ñπ [2021-12-04 23:51:19]: Annotating.</span>
<span class="co">## ‚úì [2021-12-04 23:51:19]: Annotation done.</span>
<span class="co">## ‚Ñπ [2021-12-04 23:51:19]: Summarizing per sample.</span>
<span class="co">## ‚úì [2021-12-04 23:51:19]: Summarized.</span>
<span class="co">## ‚Ñπ [2021-12-04 23:51:19]: Generating CopyNumber object.</span>
<span class="co">## ‚úì [2021-12-04 23:51:19]: Generated.</span>
<span class="co">## ‚Ñπ [2021-12-04 23:51:19]: Validating object.</span>
<span class="co">## ‚úì [2021-12-04 23:51:19]: Done.</span>
<span class="co">## ‚Ñπ [2021-12-04 23:51:19]: 0.167 secs elapsed.</span>
<span class="va">cn</span>
<span class="co">## An object of class CopyNumber </span>
<span class="co">## =============================</span>
<span class="co">##                           sample n_of_seg n_of_cnv n_of_amp n_of_del n_of_vchr</span>
<span class="co">##  1: TCGA-DF-A2KN-01A-11D-A17U-01       33        6        5        1         4</span>
<span class="co">##  2: TCGA-19-2621-01B-01D-0911-01       33        8        5        3         5</span>
<span class="co">##  3: TCGA-B6-A0X5-01A-21D-A107-01       28        8        4        4         2</span>
<span class="co">##  4: TCGA-A8-A07S-01A-11D-A036-01       38       11        2        9         4</span>
<span class="co">##  5: TCGA-26-6174-01A-21D-1842-01       43       13        8        5         8</span>
<span class="co">##  6: TCGA-CV-7432-01A-11D-2128-01       40       16        7        9         9</span>
<span class="co">##  7: TCGA-06-0644-01A-02D-0310-01       46       19        5       14         8</span>
<span class="co">##  8: TCGA-A5-A0G2-01A-11D-A042-01       39       21        5       16        10</span>
<span class="co">##  9: TCGA-99-7458-01A-11D-2035-01       48       26       10       16        13</span>
<span class="co">## 10: TCGA-05-4417-01A-22D-1854-01       52       37       33        4        17</span>
<span class="co">##     cna_burden</span>
<span class="co">##  1:      0.000</span>
<span class="co">##  2:      0.099</span>
<span class="co">##  3:      0.087</span>
<span class="co">##  4:      0.112</span>
<span class="co">##  5:      0.119</span>
<span class="co">##  6:      0.198</span>
<span class="co">##  7:      0.165</span>
<span class="co">##  8:      0.393</span>
<span class="co">##  9:      0.318</span>
<span class="co">## 10:      0.654</span></code></pre></div>
<blockquote>
<p>Currently, you can refer to <code>extract_facets_cnv()</code> and <code>extract_seqz_cnv()</code> in <a href="https://github.com/ShixiangWang/prad_signature/blob/master/analysis/src/99-functions.R" class="uri">https://github.com/ShixiangWang/prad_signature/blob/master/analysis/src/99-functions.R</a> to
see how to get tidy data from a result directory of FACETS or Sequenza.
## Tally Components</p>
</blockquote>
<p>Currently, there are two methods for generating sample-by-component matrix.</p>
<p>Option <code>sigminer.copynumber.max</code> is used to control the processing of max copy number values. Run <code><a href="https://rdrr.io/pkg/sigminer/man/sig_tally.html">?sig_tally</a></code> to see more.</p>
<div class="sourceCode" id="cb49"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="co">## Even you set max_copynumber = 20 in read_copynumber(),</span>
<span class="co">## the segmental copy number may be greater than 20</span>
<span class="co">## because for male samples, the X/Y segmental copy number</span>
<span class="co">## values will be doubled in tally process.</span>
<span class="co">## This setting will make copy number values of all segments</span>
<span class="co">## not greater than 20.</span>
<span class="fu"><a href="https://rdrr.io/r/base/options.html">options</a></span><span class="op">(</span>sigminer.copynumber.max <span class="op">=</span> <span class="fl">20</span><span class="op">)</span>
<span class="co"># Load copy number object</span>
<span class="fu"><a href="https://rdrr.io/r/base/load.html">load</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/system.file.html">system.file</a></span><span class="op">(</span><span class="st">"extdata"</span>, <span class="st">"toy_copynumber.RData"</span>,
  package <span class="op">=</span> <span class="st">"sigminer"</span>, mustWork <span class="op">=</span> <span class="cn">TRUE</span>
<span class="op">)</span><span class="op">)</span>
<span class="co"># Use method designed by Wang, Shixiang et al.</span>
<span class="va">cn_tally_W</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/pkg/sigminer/man/sig_tally.html">sig_tally</a></span><span class="op">(</span><span class="va">cn</span>, method <span class="op">=</span> <span class="st">"W"</span><span class="op">)</span></code></pre></div>
<blockquote>
<p>You can set <code>options(sigminer.sex = "male", sigminer.copynumber.max = 20)</code> at the top of your code
to avoid setting them in two places.</p>
<p><strong>Of note</strong>, the <code>sigminer.copynumber.max</code> option only has effect on <code><a href="https://rdrr.io/pkg/sigminer/man/sig_tally.html">sig_tally()</a></code> with method ‚ÄúW,‚Äù
the <code>sigminer.sex</code> option has effects on <code><a href="https://rdrr.io/pkg/sigminer/man/read_copynumber.html">read_copynumber()</a></code> and <code><a href="https://rdrr.io/pkg/sigminer/man/sig_tally.html">sig_tally()</a></code> with method ‚ÄúW.‚Äù</p>
</blockquote>
<p>This step return a <code>list</code> containing information about copy number features, components and matrix for NMF etc.</p>
</div>
<div id="extract-signatures" class="section level3" number="4.1.2">
<h3>
<span class="header-section-number">4.1.2</span> Extract signatures<a class="anchor" aria-label="anchor" href="#extract-signatures"><i class="fas fa-link"></i></a>
</h3>
<p>When you get the matrix, you can just do the signature extraction as SBS etc. signatures. So here we won‚Äôt talk much.</p>
<div class="sourceCode" id="cb50"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="va">cn_tally_W</span><span class="op">$</span><span class="va">nmf_matrix</span><span class="op">[</span><span class="fl">1</span><span class="op">:</span><span class="fl">5</span>, <span class="fl">1</span><span class="op">:</span><span class="fl">5</span><span class="op">]</span>
<span class="co">##                              BP10MB[0] BP10MB[1] BP10MB[2] BP10MB[3] BP10MB[4]</span>
<span class="co">## TCGA-05-4417-01A-22D-1854-01       275        20         5         0         0</span>
<span class="co">## TCGA-06-0644-01A-02D-0310-01       289         5         4         0         1</span>
<span class="co">## TCGA-19-2621-01B-01D-0911-01       294         2         3         1         0</span>
<span class="co">## TCGA-26-6174-01A-21D-1842-01       288         4         7         1         0</span>
<span class="co">## TCGA-99-7458-01A-11D-2035-01       284         9         5         1         1</span></code></pre></div>
<div class="sourceCode" id="cb51"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="co"># library(NMF)</span>
<span class="va">sig_w</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/pkg/sigminer/man/sig_extract.html">sig_extract</a></span><span class="op">(</span><span class="va">cn_tally_W</span><span class="op">$</span><span class="va">nmf_matrix</span>, n_sig <span class="op">=</span> <span class="fl">2</span><span class="op">)</span></code></pre></div>
</div>
</div>
</div>
  <div class="chapter-nav">
<div class="prev"><a href="basic-workflow.html"><span class="header-section-number">3</span> Basic analysis workflow</a></div>
<div class="next"><a href="target-vis.html"><span class="header-section-number">5</span> Target visualization</a></div>
</div></main><div class="col-md-3 col-lg-2 d-none d-md-block sidebar sidebar-chapter">
    <nav id="toc" data-toggle="toc" aria-label="On this page"><h2>On this page</h2>
      <ul class="nav navbar-nav">
<li><a class="nav-link" href="#analysis-supps"><span class="header-section-number">4</span> Analysis supplements</a></li>
<li>
<a class="nav-link" href="#copy-number-signature-identification"><span class="header-section-number">4.1</span> Copy number signature identification</a><ul class="nav navbar-nav">
<li><a class="nav-link" href="#read-data"><span class="header-section-number">4.1.1</span> Read data</a></li>
<li><a class="nav-link" href="#extract-signatures"><span class="header-section-number">4.1.2</span> Extract signatures</a></li>
</ul>
</li>
</ul>

      <div class="book-extra">
        <ul class="list-unstyled">
          
        </ul>
</div>
    </nav>
</div>

</div>
</div> <!-- .container -->

<footer class="bg-primary text-light mt-5"><div class="container"><div class="row">

  <div class="col-12 col-md-6 mt-3">
    <p>"<strong>Extract, Analyze and Visualize Mutational Signatures with Sigminer</strong>" was written by Shixiang Wang, PhD (Sun Yat-sen University Cancer Center), Xue-Song Liu, PhD (ShanghaiTech University). It was last built on 2021-12-04.</p>
  </div>

  <div class="col-12 col-md-6 mt-3">
    <p>This book was built by the <a class="text-light" href="https://bookdown.org">bookdown</a> R package.</p>
  </div>

</div></div>
</footer><!-- dynamically load mathjax for compatibility with self-contained --><script>
  (function () {
    var script = document.createElement("script");
    script.type = "text/javascript";
    var src = "true";
    if (src === "" || src === "true") src = "https://mathjax.rstudio.com/latest/MathJax.js?config=TeX-MML-AM_CHTML";
    if (location.protocol !== "file:")
      if (/^https?:/.test(src))
        src = src.replace(/^https?:/, '');
    script.src = src;
    document.getElementsByTagName("head")[0].appendChild(script);
  })();
</script><script type="text/x-mathjax-config">const popovers = document.querySelectorAll('a.footnote-ref[data-toggle="popover"]');
for (let popover of popovers) {
  const div = document.createElement('div');
  div.setAttribute('style', 'position: absolute; top: 0, left:0; width:0, height:0, overflow: hidden; visibility: hidden;');
  div.innerHTML = popover.getAttribute('data-content');

  var has_math = div.querySelector("span.math");
  if (has_math) {
    document.body.appendChild(div);
    MathJax.Hub.Queue(["Typeset", MathJax.Hub, div]);
    MathJax.Hub.Queue(function() {
      popover.setAttribute('data-content', div.innerHTML);
      document.body.removeChild(div);
    })
  }
}
</script>
</body>
</html>
