<!DOCTYPE html>
<html lang="en">
<head>
<meta http-equiv="Content-Type" content="text/html; charset=UTF-8">
<meta charset="utf-8">
<meta name="viewport" content="width=device-width, initial-scale=1, shrink-to-fit=no">
<title>📖 Introduction | Extract, Analyze and Visualize Mutational Signatures with Sigminer</title>
<meta name="author" content="Shixiang Wang, PhD (Sun Yat-sen University Cancer Center)">
<meta name="author" content="Xue-Song Liu, PhD (ShanghaiTech University)">
<meta name="description" content="Extract, analyze and visualize mutational signatures with sigminer.">
<meta name="generator" content="bookdown 0.24 with bs4_book()">
<meta property="og:title" content="📖 Introduction | Extract, Analyze and Visualize Mutational Signatures with Sigminer">
<meta property="og:type" content="book">
<meta property="og:description" content="Extract, analyze and visualize mutational signatures with sigminer.">
<meta name="twitter:card" content="summary">
<meta name="twitter:title" content="📖 Introduction | Extract, Analyze and Visualize Mutational Signatures with Sigminer">
<meta name="twitter:description" content="Extract, analyze and visualize mutational signatures with sigminer.">
<!-- JS --><script src="https://cdnjs.cloudflare.com/ajax/libs/clipboard.js/2.0.6/clipboard.min.js" integrity="sha256-inc5kl9MA1hkeYUt+EC3BhlIgyp/2jDIyBLS6k3UxPI=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/fuse.js/6.4.6/fuse.js" integrity="sha512-zv6Ywkjyktsohkbp9bb45V6tEMoWhzFzXis+LrMehmJZZSys19Yxf1dopHx7WzIKxr5tK2dVcYmaCk2uqdjF4A==" crossorigin="anonymous"></script><script src="https://kit.fontawesome.com/6ecbd6c532.js" crossorigin="anonymous"></script><script src="libs/header-attrs-2.10/header-attrs.js"></script><script src="libs/jquery-3.6.0/jquery-3.6.0.min.js"></script><meta name="viewport" content="width=device-width, initial-scale=1, shrink-to-fit=no">
<link href="libs/bootstrap-4.6.0/bootstrap.min.css" rel="stylesheet">
<script src="libs/bootstrap-4.6.0/bootstrap.bundle.min.js"></script><script src="libs/bs3compat-0.3.0/transition.js"></script><script src="libs/bs3compat-0.3.0/tabs.js"></script><script src="libs/bs3compat-0.3.0/bs3compat.js"></script><link href="libs/bs4_book-1.0.0/bs4_book.css" rel="stylesheet">
<script src="libs/bs4_book-1.0.0/bs4_book.js"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/autocomplete.js/0.38.0/autocomplete.jquery.min.js" integrity="sha512-GU9ayf+66Xx2TmpxqJpliWbT5PiGYxpaG8rfnBEk1LL8l1KGkRShhngwdXK1UgqhAzWpZHSiYPc09/NwDQIGyg==" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/mark.js/8.11.1/mark.min.js" integrity="sha512-5CYOlHXGh6QpOFA/TeTylKLWfB3ftPsde7AnmhuitiTX4K5SqCLBeKro6sPS8ilsz1Q4NRx3v8Ko2IBiszzdww==" crossorigin="anonymous"></script><!-- CSS -->
</head>
<body data-spy="scroll" data-target="#toc">

<div class="container-fluid">
<div class="row">
  <header class="col-sm-12 col-lg-3 sidebar sidebar-book"><a class="sr-only sr-only-focusable" href="#content">Skip to main content</a>

    <div class="d-flex align-items-start justify-content-between">
      <h1>
        <a href="index.html" title="">Extract, Analyze and Visualize Mutational Signatures with Sigminer</a>
      </h1>
      <button class="btn btn-outline-primary d-lg-none ml-2 mt-1" type="button" data-toggle="collapse" data-target="#main-nav" aria-expanded="true" aria-controls="main-nav"><i class="fas fa-bars"></i><span class="sr-only">Show table of contents</span></button>
    </div>

    <div id="main-nav" class="collapse-lg">
      <form role="search">
        <input id="search" class="form-control" type="search" placeholder="Search" aria-label="Search">
</form>

      <nav aria-label="Table of contents"><h2>Table of contents</h2>
        <ul class="book-toc list-unstyled">
<li><a class="active" href="index.html">📖 Introduction</a></li>
<li class="book-part">Part I: Background and Prerequisite</li>
<li><a class="" href="mutsig-intro.html"><span class="header-section-number">1</span> Mutational signatures</a></li>
<li><a class="" href="package-prerequisite-and-installation.html"><span class="header-section-number">2</span> Package prerequisite and installation</a></li>
<li class="book-part">Part II: Workflows</li>
<li class="book-part">Part III: Miscellaneous topics</li>
<li class="book-part">Appendix</li>
<li><a class="" href="references.html">References</a></li>
</ul>

        <div class="book-extra">
          
        </div>
      </nav>
</div>
  </header><main class="col-sm-12 col-md-9 col-lg-7" id="content"><!--bookdown:title:end--><!--bookdown:title:start--><div id="introduction" class="section level1 unnumbered">
<h1>📖 Introduction<a class="anchor" aria-label="anchor" href="#introduction"><i class="fas fa-link"></i></a>
</h1>
<p><a href="%5Bsigminer%5D(https://CRAN.R-project.org/package=sigminer)"><img src="https://img.shields.io/badge/sigminer-2.0.6-green.svg"></a>
<a href="%5BUCSCXenaTools%5D(https://CRAN.R-project.org/package=UCSCXenaTools)"><img src="https://img.shields.io/badge/UCSCXenaTools-1.4.6-green.svg"></a>
<a href="%5Bmaftools%5D(http://bioconductor.org/packages/maftools)"><img src="https://img.shields.io/badge/maftools-2.8.0-green.svg"></a>
<a href="https://hits.seeyoufarm.com"><img src="https://hits.seeyoufarm.com/api/count/incr/badge.svg?url=https%3A%2F%2Fgithub.com%2FShixiangWang%2Fsigminer-book&amp;count_bg=%2379C83D&amp;title_bg=%23555555&amp;icon=americanairlines.svg&amp;icon_color=%23E7E7E7&amp;title=hits&amp;edge_flat=false" alt="Hits"></a></p>
<div id="motivation" class="section level2 unnumbered">
<h2>🎯 Motivation<a class="anchor" aria-label="anchor" href="#motivation"><i class="fas fa-link"></i></a>
</h2>
<p>The book is written as a guide for extracting, analyzing and visualizing mutational signatures with R/CRAN package <strong>sigminer</strong>. The <a href="https://github.com/ShixiangWang/sigminer/blob/master/README.md"><em>README</em></a> and <a href="https://shixiangwang.github.io/sigminer/reference/index.html"><em>Reference list</em></a> of <strong>sigminer</strong> have given users overview and the very details of specific points (e.g., functions) in <strong>sigminer</strong>. This book will help users focus on quickly getting the mutational signature analysis done to make life easy.</p>
<p>In this book, we assume you have already known how to operate <a href="http://cran.r-project.org/">R</a>.</p>
</div>
<div id="citation" class="section level2 unnumbered">
<h2>📝 Citation<a class="anchor" aria-label="anchor" href="#citation"><i class="fas fa-link"></i></a>
</h2>
<p>If you use <strong>sigminer</strong> or its pipeline version <strong>sigflow</strong> in published research, please cite the most appropriate paper(s):</p>
<ol style="list-style-type: decimal">
<li>
<strong>Wang, S.</strong>, Li, H., Song, M., Tao, Z., Wu, T., He, Z., … &amp; Liu, X. S. (2021). Copy number signature analysis tool and its application in prostate cancer reveals distinct mutational processes and clinical outcomes. <strong><em>PLoS genetics</em></strong>, 17(5), e1009557. <a href="https://doi.org/10.1371/journal.pgen.1009557" class="uri">https://doi.org/10.1371/journal.pgen.1009557</a>
</li>
<li>
<strong>Wang, S.</strong>, Tao, Z., Wu, T., &amp; Liu, X. S. (2021). Sigflow: an automated and comprehensive pipeline for cancer genome mutational signature analysis. <strong><em>Bioinformatics</em></strong>, 37(11), 1590-1592. <a href="https://doi.org/10.1093/bioinformatics/btaa895" class="uri">https://doi.org/10.1093/bioinformatics/btaa895</a>
</li>
</ol>
<p>Please properly cite the following references when you are using any
corresponding features. The references are also listed in the function
documentation. Very thanks to the works, <strong>sigminer</strong> cannot be created
without the giants.</p>
<ol style="list-style-type: decimal">
<li>Mayakonda, Anand, et al. “Maftools: efficient and comprehensive
analysis of somatic variants in cancer.” <strong><em>Genome research</em></strong> 28.11
(2018): 1747-1756.</li>
<li>Gaujoux, Renaud, and Cathal Seoighe. “A Flexible R Package for
Nonnegative Matrix Factorization.”" <strong><em>BMC Bioinformatics</em></strong> 11, no. 1
(December 2010).</li>
<li>Wickham, Hadley. “ggplot2.” <strong><em>Wiley Interdisciplinary Reviews: Computational Statistics</em></strong> 3.2 (2011): 180-185.</li>
<li>Kim, Jaegil, et al. “Somatic ERCC2 mutations are associated with a
distinct genomic signature in urothelial tumors.” <strong><em>Nature Genetics</em></strong>
48.6 (2016): 600.</li>
<li>Alexandrov, Ludmil B., et al. “Deciphering signatures of mutational
processes operative in human cancer.” <strong><em>Cell Reports</em></strong> 3.1 (2013):
246-259.</li>
<li>Degasperi, Andrea, et al. “A practical framework and online tool for
mutational signature analyses show intertissue variation and driver
dependencies.” <strong><em>Nature Cancer</em></strong> 1.2 (2020): 249-263.</li>
<li>Alexandrov, Ludmil B., et al. “The repertoire of mutational
signatures in human cancer.” <strong><em>Nature</em></strong> 578.7793 (2020): 94-101.</li>
<li>Macintyre, Geoff, et al. “Copy number signatures and mutational
processes in ovarian carcinoma.” <strong><em>Nature Genetics</em></strong> 50.9 (2018): 1262.</li>
<li>Tan, Vincent YF, and Cédric Févotte. “Automatic relevance
determination in nonnegative matrix factorization with the/spl
beta/-divergence.” <strong><em>IEEE Transactions on Pattern Analysis and Machine Intelligence</em></strong> 35.7 (2012): 1592-1605.</li>
<li>Bergstrom EN, Huang MN, Mahto U, Barnes M, Stratton MR, Rozen SG,
Alexandrov LB: SigProfilerMatrixGenerator: a tool for visualizing
and exploring patterns of small mutational events. <strong><em>BMC Genomics</em></strong>
2019, 20:685
<a href="https://bmcgenomics.biomedcentral.com/articles/10.1186/s12864-019-6041-2" class="uri">https://bmcgenomics.biomedcentral.com/articles/10.1186/s12864-019-6041-2</a>
</li>
</ol>
</div>
<div id="book-structure" class="section level2 unnumbered">
<h2>📚 Book structure<a class="anchor" aria-label="anchor" href="#book-structure"><i class="fas fa-link"></i></a>
</h2>
<ul>
<li>Part 1 (Background and Prerequisite) describes the basic concepts of mutational signature and how to install/load <strong>sigminer</strong>.</li>
<li>Part 2 (Workflows) introduces how to prepare your input data and run mutational signature analysis for different mutation data types (SBS, DBS, INDEL, Genome rearrangement, CNV) with different methods provided by <strong>sigminer</strong>.</li>
<li>Part 3 (Miscellaneous topics) describes useful utilities including builtin datasets, SBS signature conversion.</li>
</ul>
</div>
<div id="want-to-help" class="section level2 unnumbered">
<h2>💖 Want to help?<a class="anchor" aria-label="anchor" href="#want-to-help"><i class="fas fa-link"></i></a>
</h2>
<p>The book’s source code is hosted on GitHub, at <a href="https://github.com/ShixiangWang/sigminer-book" class="uri">https://github.com/ShixiangWang/sigminer-book</a>. Any feedback on the book is very welcome. Feel free to <a href="https://github.com/ShixiangWang/sigminer-book/issues/new">open an issue</a> on GitHub or send me a pull request if you notice typos or other issues (I’m not a native English speaker ;) ).</p>
</div>
<div id="bug-report-or-feature-request" class="section level2 unnumbered">
<h2>🐜 Bug report or feature request<a class="anchor" aria-label="anchor" href="#bug-report-or-feature-request"><i class="fas fa-link"></i></a>
</h2>
<p>If you find any bugs or want to have a new feature, please <a href="https://github.com/ShixiangWang/sigminer/issues/new">file an issue</a>.</p>
</div>
<div id="acknowlegment" class="section level2 unnumbered">
<h2>🌵 Acknowlegment<a class="anchor" aria-label="anchor" href="#acknowlegment"><i class="fas fa-link"></i></a>
</h2>
<p>I built this book website by imitating <a href="https://github.com/YuLab-SMU/biomedical-knowledge-mining-book"><em>Biomedical knowledge mining using GOSemSim and clusterProfiler</em></a> and reusing its configurations, I would like to thank <a href="https://github.com/GuangchuangYu">Guangchuang Yu</a> here.</p>

</div>
</div>



  <div class="chapter-nav">
<div class="empty"></div>
<div class="next"><a href="mutsig-intro.html"><span class="header-section-number">1</span> Mutational signatures</a></div>
</div></main><div class="col-md-3 col-lg-2 d-none d-md-block sidebar sidebar-chapter">
    <nav id="toc" data-toggle="toc" aria-label="On this page"><h2>On this page</h2>
      <ul class="nav navbar-nav">
<li><a class="nav-link" href="#introduction">📖 Introduction</a></li>
<li><a class="nav-link" href="#motivation">🎯 Motivation</a></li>
<li><a class="nav-link" href="#citation">📝 Citation</a></li>
<li><a class="nav-link" href="#book-structure">📚 Book structure</a></li>
<li><a class="nav-link" href="#want-to-help">💖 Want to help?</a></li>
<li><a class="nav-link" href="#bug-report-or-feature-request">🐜 Bug report or feature request</a></li>
<li><a class="nav-link" href="#acknowlegment">🌵 Acknowlegment</a></li>
</ul>

      <div class="book-extra">
        <ul class="list-unstyled">
          
        </ul>
</div>
    </nav>
</div>

</div>
</div> <!-- .container -->

<footer class="bg-primary text-light mt-5"><div class="container"><div class="row">

  <div class="col-12 col-md-6 mt-3">
    <p>"<strong>Extract, Analyze and Visualize Mutational Signatures with Sigminer</strong>" was written by Shixiang Wang, PhD (Sun Yat-sen University Cancer Center), Xue-Song Liu, PhD (ShanghaiTech University). It was last built on 2021-09-05.</p>
  </div>

  <div class="col-12 col-md-6 mt-3">
    <p>This book was built by the <a class="text-light" href="https://bookdown.org">bookdown</a> R package.</p>
  </div>

</div></div>
</footer>
</body>
</html>
